{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {},
     "inputWidgets": {},
     "nuid": "d3c0a68e-950d-421c-a5bf-d94eb7e96a2e",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "UsageError: Line magic function `%fs` not found.\n"
     ]
    }
   ],
   "source": [
    "%fs ls file:/Workspace/Repos/shared/beacon2-ri-tools/CINECA_synthetic_cohort_EUROPE_UK1/Beacon-v2-Models_CINECA_UK1.xlsx"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {},
     "inputWidgets": {},
     "nuid": "fe564e4e-d2e5-4184-9b15-ed9117f503d1",
     "showTitle": false,
     "title": ""
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: pip in /Users/marionfs/jupyter-venv/lib/python3.8/site-packages (23.0)\n",
      "Collecting pip\n",
      "  Downloading pip-23.0.1-py3-none-any.whl (2.1 MB)\n",
      "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m2.1/2.1 MB\u001b[0m \u001b[31m3.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m00:01\u001b[0m00:01\u001b[0m\n",
      "\u001b[?25hInstalling collected packages: pip\n",
      "  Attempting uninstall: pip\n",
      "    Found existing installation: pip 23.0\n",
      "    Uninstalling pip-23.0:\n",
      "      Successfully uninstalled pip-23.0\n",
      "Successfully installed pip-23.0.1\n",
      "Requirement already satisfied: openpyxl in /Users/marionfs/jupyter-venv/lib/python3.8/site-packages (3.0.10)\n",
      "Requirement already satisfied: et-xmlfile in /Users/marionfs/jupyter-venv/lib/python3.8/site-packages (from openpyxl) (1.1.0)\n"
     ]
    }
   ],
   "source": [
    "!python -m pip install --upgrade pip\n",
    "!pip install openpyxl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {},
     "inputWidgets": {},
     "nuid": "91563b26-f032-4e01-ac70-4a736155179a",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import json\n",
    "import random\n",
    "import ast\n",
    "import pprint"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {},
     "inputWidgets": {},
     "nuid": "90a30b26-2a46-4115-a498-a27c97e1afbf",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [],
   "source": [
    "work_dir = '/Users/marionfs/Documents/GitHub/sbeacon-exploration/data/CINECA_synthetic_cohort_EUROPE_UK1/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {},
     "inputWidgets": {},
     "nuid": "ea7959b7-cfe2-4f7d-8a1a-03a52240614a",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [],
   "source": [
    "submission_json = {\n",
    "    \"index\": True,\n",
    "\t\"datasetId\": \"CINECA_UMCCR\",\n",
    "\t\"dataset\": {\n",
    "\t\t\"name\": \"CINECA_UMCCR_NAME\",\n",
    "\t\t\"description\": \"Test CINECA dataset for UMCCR\"\n",
    "\t},\n",
    "\t\"assemblyId\": \"GRCh38\",\n",
    "\t\"cohortId\": \"TestCohortCineca1\",\n",
    "\t\"cohort\": {\n",
    "\t\t\"name\": \"UMCCR CINECA test cohort 1\",\n",
    "\t\t\"cohortType\": \"study-defined\"\n",
    "\t},\n",
    "\t\"vcfLocations\": [\n",
    "\t\t\"s3://org.umccr.onboarding.testdata/CINECA_UK1/Test.1000G.phase3.joint.lifted.vcf.gz\"\n",
    "\t],\n",
    "\t\"individuals\": [],\n",
    "\t\"biosamples\": [],\n",
    "\t\"analyses\": [],\n",
    "    \"runs\": []\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {},
     "inputWidgets": {},
     "nuid": "030fb607-0327-4ce8-b167-db1b79bd340b",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/marionfs/jupyter-venv/lib/python3.8/site-packages/openpyxl/worksheet/_reader.py:312: UserWarning: Unknown extension is not supported and will be removed\n",
      "  warn(msg)\n"
     ]
    }
   ],
   "source": [
    "df = pd.read_excel(f'{work_dir}/Beacon-v2-Models_CINECA_UK1MS.xlsx', sheet_name='individuals')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {},
     "inputWidgets": {},
     "nuid": "f584e7af-bc3f-49eb-9bf6-123256849703",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>diseases_ageOfOnset</th>\n",
       "      <th>diseases_diseaseCode.id</th>\n",
       "      <th>diseases_diseaseCode.label</th>\n",
       "      <th>diseases_familyHistory</th>\n",
       "      <th>diseases_notes</th>\n",
       "      <th>diseases_severity.id</th>\n",
       "      <th>diseases_severity.label</th>\n",
       "      <th>diseases_stage.id</th>\n",
       "      <th>diseases_stage.label</th>\n",
       "      <th>ethnicity.id</th>\n",
       "      <th>...</th>\n",
       "      <th>treatments_cumulativeDose.referenceRange.low</th>\n",
       "      <th>treatments_cumulativeDose.referenceRange.unit</th>\n",
       "      <th>treatments_cumulativeDose.unit.id</th>\n",
       "      <th>treatments_cumulativeDose.unit.label</th>\n",
       "      <th>treatments_cumulativeDose.value</th>\n",
       "      <th>treatments_doseIntervals</th>\n",
       "      <th>treatments_routeOfAdministration.id</th>\n",
       "      <th>treatments_routeOfAdministration.label</th>\n",
       "      <th>treatments_treatmentCode.id</th>\n",
       "      <th>treatments_treatmentCode.label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NCIT:C42331</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NCIT:C41261</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NCIT:C41260</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NCIT:C67109</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NCIT:C67109</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2499</th>\n",
       "      <td>NaN</td>\n",
       "      <td>MONDO:0001609,MONDO:0004126</td>\n",
       "      <td>agranulocytosis,thyroiditis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NCIT:C43856</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2500</th>\n",
       "      <td>NaN</td>\n",
       "      <td>MONDO:0019338,MONDO:0005155,MONDO:0100430</td>\n",
       "      <td>sarcoidosis,cirrhosis of liver,fibrotic liver ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NCIT:C67109</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2501</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NCIT:C67109</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2502</th>\n",
       "      <td>NaN</td>\n",
       "      <td>MONDO:0004126,MONDO:0007186</td>\n",
       "      <td>thyroiditis,gastroesophageal reflux disease</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2503</th>\n",
       "      <td>NaN</td>\n",
       "      <td>MONDO:0005155,MONDO:0100430</td>\n",
       "      <td>cirrhosis of liver,fibrotic liver disease</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NCIT:C41260</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2504 rows × 82 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      diseases_ageOfOnset                    diseases_diseaseCode.id  \\\n",
       "0                     NaN                                        NaN   \n",
       "1                     NaN                                        NaN   \n",
       "2                     NaN                                        NaN   \n",
       "3                     NaN                                        NaN   \n",
       "4                     NaN                                        NaN   \n",
       "...                   ...                                        ...   \n",
       "2499                  NaN                MONDO:0001609,MONDO:0004126   \n",
       "2500                  NaN  MONDO:0019338,MONDO:0005155,MONDO:0100430   \n",
       "2501                  NaN                                        NaN   \n",
       "2502                  NaN                MONDO:0004126,MONDO:0007186   \n",
       "2503                  NaN                MONDO:0005155,MONDO:0100430   \n",
       "\n",
       "                             diseases_diseaseCode.label  \\\n",
       "0                                                   NaN   \n",
       "1                                                   NaN   \n",
       "2                                                   NaN   \n",
       "3                                                   NaN   \n",
       "4                                                   NaN   \n",
       "...                                                 ...   \n",
       "2499                        agranulocytosis,thyroiditis   \n",
       "2500  sarcoidosis,cirrhosis of liver,fibrotic liver ...   \n",
       "2501                                                NaN   \n",
       "2502        thyroiditis,gastroesophageal reflux disease   \n",
       "2503          cirrhosis of liver,fibrotic liver disease   \n",
       "\n",
       "      diseases_familyHistory  diseases_notes  diseases_severity.id  \\\n",
       "0                        NaN             NaN                   NaN   \n",
       "1                        NaN             NaN                   NaN   \n",
       "2                        NaN             NaN                   NaN   \n",
       "3                        NaN             NaN                   NaN   \n",
       "4                        NaN             NaN                   NaN   \n",
       "...                      ...             ...                   ...   \n",
       "2499                     NaN             NaN                   NaN   \n",
       "2500                     NaN             NaN                   NaN   \n",
       "2501                     NaN             NaN                   NaN   \n",
       "2502                     NaN             NaN                   NaN   \n",
       "2503                     NaN             NaN                   NaN   \n",
       "\n",
       "      diseases_severity.label  diseases_stage.id  diseases_stage.label  \\\n",
       "0                         NaN                NaN                   NaN   \n",
       "1                         NaN                NaN                   NaN   \n",
       "2                         NaN                NaN                   NaN   \n",
       "3                         NaN                NaN                   NaN   \n",
       "4                         NaN                NaN                   NaN   \n",
       "...                       ...                ...                   ...   \n",
       "2499                      NaN                NaN                   NaN   \n",
       "2500                      NaN                NaN                   NaN   \n",
       "2501                      NaN                NaN                   NaN   \n",
       "2502                      NaN                NaN                   NaN   \n",
       "2503                      NaN                NaN                   NaN   \n",
       "\n",
       "     ethnicity.id  ... treatments_cumulativeDose.referenceRange.low  \\\n",
       "0     NCIT:C42331  ...                                          NaN   \n",
       "1     NCIT:C41261  ...                                          NaN   \n",
       "2     NCIT:C41260  ...                                          NaN   \n",
       "3     NCIT:C67109  ...                                          NaN   \n",
       "4     NCIT:C67109  ...                                          NaN   \n",
       "...           ...  ...                                          ...   \n",
       "2499  NCIT:C43856  ...                                          NaN   \n",
       "2500  NCIT:C67109  ...                                          NaN   \n",
       "2501  NCIT:C67109  ...                                          NaN   \n",
       "2502          NaN  ...                                          NaN   \n",
       "2503  NCIT:C41260  ...                                          NaN   \n",
       "\n",
       "      treatments_cumulativeDose.referenceRange.unit  \\\n",
       "0                                               NaN   \n",
       "1                                               NaN   \n",
       "2                                               NaN   \n",
       "3                                               NaN   \n",
       "4                                               NaN   \n",
       "...                                             ...   \n",
       "2499                                            NaN   \n",
       "2500                                            NaN   \n",
       "2501                                            NaN   \n",
       "2502                                            NaN   \n",
       "2503                                            NaN   \n",
       "\n",
       "      treatments_cumulativeDose.unit.id  treatments_cumulativeDose.unit.label  \\\n",
       "0                                   NaN                                   NaN   \n",
       "1                                   NaN                                   NaN   \n",
       "2                                   NaN                                   NaN   \n",
       "3                                   NaN                                   NaN   \n",
       "4                                   NaN                                   NaN   \n",
       "...                                 ...                                   ...   \n",
       "2499                                NaN                                   NaN   \n",
       "2500                                NaN                                   NaN   \n",
       "2501                                NaN                                   NaN   \n",
       "2502                                NaN                                   NaN   \n",
       "2503                                NaN                                   NaN   \n",
       "\n",
       "      treatments_cumulativeDose.value  treatments_doseIntervals  \\\n",
       "0                                 NaN                       NaN   \n",
       "1                                 NaN                       NaN   \n",
       "2                                 NaN                       NaN   \n",
       "3                                 NaN                       NaN   \n",
       "4                                 NaN                       NaN   \n",
       "...                               ...                       ...   \n",
       "2499                              NaN                       NaN   \n",
       "2500                              NaN                       NaN   \n",
       "2501                              NaN                       NaN   \n",
       "2502                              NaN                       NaN   \n",
       "2503                              NaN                       NaN   \n",
       "\n",
       "      treatments_routeOfAdministration.id  \\\n",
       "0                                     NaN   \n",
       "1                                     NaN   \n",
       "2                                     NaN   \n",
       "3                                     NaN   \n",
       "4                                     NaN   \n",
       "...                                   ...   \n",
       "2499                                  NaN   \n",
       "2500                                  NaN   \n",
       "2501                                  NaN   \n",
       "2502                                  NaN   \n",
       "2503                                  NaN   \n",
       "\n",
       "      treatments_routeOfAdministration.label  treatments_treatmentCode.id  \\\n",
       "0                                        NaN                          NaN   \n",
       "1                                        NaN                          NaN   \n",
       "2                                        NaN                          NaN   \n",
       "3                                        NaN                          NaN   \n",
       "4                                        NaN                          NaN   \n",
       "...                                      ...                          ...   \n",
       "2499                                     NaN                          NaN   \n",
       "2500                                     NaN                          NaN   \n",
       "2501                                     NaN                          NaN   \n",
       "2502                                     NaN                          NaN   \n",
       "2503                                     NaN                          NaN   \n",
       "\n",
       "     treatments_treatmentCode.label  \n",
       "0                               NaN  \n",
       "1                               NaN  \n",
       "2                               NaN  \n",
       "3                               NaN  \n",
       "4                               NaN  \n",
       "...                             ...  \n",
       "2499                            NaN  \n",
       "2500                            NaN  \n",
       "2501                            NaN  \n",
       "2502                            NaN  \n",
       "2503                            NaN  \n",
       "\n",
       "[2504 rows x 82 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {},
     "inputWidgets": {},
     "nuid": "47b3a11f-bcc5-4dcb-9daf-899a06e40838",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['diseases_ageOfOnset', 'diseases_diseaseCode.id',\n",
       "       'diseases_diseaseCode.label', 'diseases_familyHistory',\n",
       "       'diseases_notes', 'diseases_severity.id', 'diseases_severity.label',\n",
       "       'diseases_stage.id', 'diseases_stage.label', 'ethnicity.id',\n",
       "       'ethnicity.label', 'exposures_ageAtExposure.iso8601duration',\n",
       "       'exposures_date', 'exposures_duration', 'exposures_exposureCode.id',\n",
       "       'exposures_exposureCode.label', 'exposures_unit.id',\n",
       "       'exposures_unit.label', 'exposures_value', 'geographicOrigin.id',\n",
       "       'geographicOrigin.label', 'id', 'info.eid',\n",
       "       'interventionsOrProcedures_ageAtProcedure',\n",
       "       'interventionsOrProcedures_bodySite.id',\n",
       "       'interventionsOrProcedures_bodySite.label',\n",
       "       'interventionsOrProcedures_dateOfProcedure',\n",
       "       'interventionsOrProcedures_procedureCode.id',\n",
       "       'interventionsOrProcedures_procedureCode.label', 'karyotypicSex',\n",
       "       'measures_assayCode.id', 'measures_assayCode.label', 'measures_date',\n",
       "       'measures_measurementValue', 'measures_notes',\n",
       "       'measures_observationMoment', 'measures_procedure.ageAtProcedure',\n",
       "       'measures_procedure.bodySite.id', 'measures_procedure.bodySite.label',\n",
       "       'measures_procedure.dateOfProcedure',\n",
       "       'measures_procedure.procedureCode.id',\n",
       "       'measures_procedure.procedureCode.label',\n",
       "       'pedigrees_disease.ageOfOnset', 'pedigrees_disease.diseaseCode.id',\n",
       "       'pedigrees_disease.diseaseCode.label',\n",
       "       'pedigrees_disease.familyHistory', 'pedigrees_disease.notes',\n",
       "       'pedigrees_disease.severity.id', 'pedigrees_disease.severity.label',\n",
       "       'pedigrees_disease.stage.id', 'pedigrees_disease.stage.label',\n",
       "       'pedigrees_id', 'pedigrees_members', 'pedigrees_numSubjects',\n",
       "       'phenotypicFeatures_evidence.evidenceCode.id',\n",
       "       'phenotypicFeatures_evidence.evidenceCode.label',\n",
       "       'phenotypicFeatures_evidence.reference.id',\n",
       "       'phenotypicFeatures_evidence.reference.notes',\n",
       "       'phenotypicFeatures_evidence.reference.reference',\n",
       "       'phenotypicFeatures_excluded', 'phenotypicFeatures_featureType.id',\n",
       "       'phenotypicFeatures_featureType.label', 'phenotypicFeatures_modifiers',\n",
       "       'phenotypicFeatures_notes', 'phenotypicFeatures_onset',\n",
       "       'phenotypicFeatures_resolution', 'phenotypicFeatures_severity.id',\n",
       "       'phenotypicFeatures_severity.label', 'sex.id', 'sex.label',\n",
       "       'treatments_ageAtOnset.iso8601duration',\n",
       "       'treatments_cumulativeDose.referenceRange.high',\n",
       "       'treatments_cumulativeDose.referenceRange.low',\n",
       "       'treatments_cumulativeDose.referenceRange.unit',\n",
       "       'treatments_cumulativeDose.unit.id',\n",
       "       'treatments_cumulativeDose.unit.label',\n",
       "       'treatments_cumulativeDose.value', 'treatments_doseIntervals',\n",
       "       'treatments_routeOfAdministration.id',\n",
       "       'treatments_routeOfAdministration.label', 'treatments_treatmentCode.id',\n",
       "       'treatments_treatmentCode.label'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {},
     "inputWidgets": {},
     "nuid": "93c09465-916f-4b1b-9d05-f4d66382b485",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>info.eid</th>\n",
       "      <th>diseases_diseaseCode.id</th>\n",
       "      <th>diseases_diseaseCode.label</th>\n",
       "      <th>diseases_familyHistory</th>\n",
       "      <th>diseases_notes</th>\n",
       "      <th>diseases_ageOfOnset</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HG00096</td>\n",
       "      <td>fake1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HG00097</td>\n",
       "      <td>fake2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HG00099</td>\n",
       "      <td>fake3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HG00100</td>\n",
       "      <td>fake4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HG00101</td>\n",
       "      <td>fake5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2499</th>\n",
       "      <td>NA21137</td>\n",
       "      <td>fake2500</td>\n",
       "      <td>MONDO:0001609,MONDO:0004126</td>\n",
       "      <td>agranulocytosis,thyroiditis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2500</th>\n",
       "      <td>NA21141</td>\n",
       "      <td>fake2501</td>\n",
       "      <td>MONDO:0019338,MONDO:0005155,MONDO:0100430</td>\n",
       "      <td>sarcoidosis,cirrhosis of liver,fibrotic liver ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2501</th>\n",
       "      <td>NA21142</td>\n",
       "      <td>fake2502</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2502</th>\n",
       "      <td>NA21143</td>\n",
       "      <td>fake2503</td>\n",
       "      <td>MONDO:0004126,MONDO:0007186</td>\n",
       "      <td>thyroiditis,gastroesophageal reflux disease</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2503</th>\n",
       "      <td>NA21144</td>\n",
       "      <td>fake2504</td>\n",
       "      <td>MONDO:0005155,MONDO:0100430</td>\n",
       "      <td>cirrhosis of liver,fibrotic liver disease</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2504 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           id  info.eid                    diseases_diseaseCode.id  \\\n",
       "0     HG00096     fake1                                        NaN   \n",
       "1     HG00097     fake2                                        NaN   \n",
       "2     HG00099     fake3                                        NaN   \n",
       "3     HG00100     fake4                                        NaN   \n",
       "4     HG00101     fake5                                        NaN   \n",
       "...       ...       ...                                        ...   \n",
       "2499  NA21137  fake2500                MONDO:0001609,MONDO:0004126   \n",
       "2500  NA21141  fake2501  MONDO:0019338,MONDO:0005155,MONDO:0100430   \n",
       "2501  NA21142  fake2502                                        NaN   \n",
       "2502  NA21143  fake2503                MONDO:0004126,MONDO:0007186   \n",
       "2503  NA21144  fake2504                MONDO:0005155,MONDO:0100430   \n",
       "\n",
       "                             diseases_diseaseCode.label  \\\n",
       "0                                                   NaN   \n",
       "1                                                   NaN   \n",
       "2                                                   NaN   \n",
       "3                                                   NaN   \n",
       "4                                                   NaN   \n",
       "...                                                 ...   \n",
       "2499                        agranulocytosis,thyroiditis   \n",
       "2500  sarcoidosis,cirrhosis of liver,fibrotic liver ...   \n",
       "2501                                                NaN   \n",
       "2502        thyroiditis,gastroesophageal reflux disease   \n",
       "2503          cirrhosis of liver,fibrotic liver disease   \n",
       "\n",
       "      diseases_familyHistory  diseases_notes  diseases_ageOfOnset  \n",
       "0                        NaN             NaN                  NaN  \n",
       "1                        NaN             NaN                  NaN  \n",
       "2                        NaN             NaN                  NaN  \n",
       "3                        NaN             NaN                  NaN  \n",
       "4                        NaN             NaN                  NaN  \n",
       "...                      ...             ...                  ...  \n",
       "2499                     NaN             NaN                  NaN  \n",
       "2500                     NaN             NaN                  NaN  \n",
       "2501                     NaN             NaN                  NaN  \n",
       "2502                     NaN             NaN                  NaN  \n",
       "2503                     NaN             NaN                  NaN  \n",
       "\n",
       "[2504 rows x 7 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_disease = df[['id', 'info.eid', 'diseases_diseaseCode.id', 'diseases_diseaseCode.label', 'diseases_familyHistory', 'diseases_notes', 'diseases_ageOfOnset']]\n",
    "df_disease"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {},
     "inputWidgets": {},
     "nuid": "da7cce06-41a5-4f7d-9e48-fa87a2f83374",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>info.eid</th>\n",
       "      <th>karyotypicSex</th>\n",
       "      <th>sex.id</th>\n",
       "      <th>sex.label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HG00096</td>\n",
       "      <td>fake1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NCIT:C20197</td>\n",
       "      <td>male</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HG00097</td>\n",
       "      <td>fake2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NCIT:C16576</td>\n",
       "      <td>female</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HG00099</td>\n",
       "      <td>fake3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NCIT:C16576</td>\n",
       "      <td>female</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HG00100</td>\n",
       "      <td>fake4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NCIT:C16576</td>\n",
       "      <td>female</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HG00101</td>\n",
       "      <td>fake5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NCIT:C20197</td>\n",
       "      <td>male</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2499</th>\n",
       "      <td>NA21137</td>\n",
       "      <td>fake2500</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NCIT:C16576</td>\n",
       "      <td>female</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2500</th>\n",
       "      <td>NA21141</td>\n",
       "      <td>fake2501</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NCIT:C16576</td>\n",
       "      <td>female</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2501</th>\n",
       "      <td>NA21142</td>\n",
       "      <td>fake2502</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NCIT:C16576</td>\n",
       "      <td>female</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2502</th>\n",
       "      <td>NA21143</td>\n",
       "      <td>fake2503</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NCIT:C16576</td>\n",
       "      <td>female</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2503</th>\n",
       "      <td>NA21144</td>\n",
       "      <td>fake2504</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NCIT:C16576</td>\n",
       "      <td>female</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2504 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           id  info.eid  karyotypicSex       sex.id sex.label\n",
       "0     HG00096     fake1            NaN  NCIT:C20197      male\n",
       "1     HG00097     fake2            NaN  NCIT:C16576    female\n",
       "2     HG00099     fake3            NaN  NCIT:C16576    female\n",
       "3     HG00100     fake4            NaN  NCIT:C16576    female\n",
       "4     HG00101     fake5            NaN  NCIT:C20197      male\n",
       "...       ...       ...            ...          ...       ...\n",
       "2499  NA21137  fake2500            NaN  NCIT:C16576    female\n",
       "2500  NA21141  fake2501            NaN  NCIT:C16576    female\n",
       "2501  NA21142  fake2502            NaN  NCIT:C16576    female\n",
       "2502  NA21143  fake2503            NaN  NCIT:C16576    female\n",
       "2503  NA21144  fake2504            NaN  NCIT:C16576    female\n",
       "\n",
       "[2504 rows x 5 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_sex = df[['id', 'info.eid', 'karyotypicSex', 'sex.id', 'sex.label']]\n",
    "df_sex"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {},
     "inputWidgets": {},
     "nuid": "0ab23c31-5903-4be8-a353-da9c9513605f",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [],
   "source": [
    "df_ethnicity = df[['id', 'info.eid', 'ethnicity.id', 'ethnicity.label']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {},
     "inputWidgets": {},
     "nuid": "82d5cd09-a907-479a-a754-fc01c2273f92",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>info.eid</th>\n",
       "      <th>geographicOrigin.id</th>\n",
       "      <th>geographicOrigin.label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HG00096</td>\n",
       "      <td>fake1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HG00097</td>\n",
       "      <td>fake2</td>\n",
       "      <td>GAZ:00002641</td>\n",
       "      <td>England</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HG00099</td>\n",
       "      <td>fake3</td>\n",
       "      <td>GAZ:00002640</td>\n",
       "      <td>Wales</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HG00100</td>\n",
       "      <td>fake4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HG00101</td>\n",
       "      <td>fake5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2499</th>\n",
       "      <td>NA21137</td>\n",
       "      <td>fake2500</td>\n",
       "      <td>GAZ:00002638</td>\n",
       "      <td>Northern Ireland</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2500</th>\n",
       "      <td>NA21141</td>\n",
       "      <td>fake2501</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2501</th>\n",
       "      <td>NA21142</td>\n",
       "      <td>fake2502</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2502</th>\n",
       "      <td>NA21143</td>\n",
       "      <td>fake2503</td>\n",
       "      <td>GAZ:00002638</td>\n",
       "      <td>Northern Ireland</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2503</th>\n",
       "      <td>NA21144</td>\n",
       "      <td>fake2504</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2504 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           id  info.eid geographicOrigin.id geographicOrigin.label\n",
       "0     HG00096     fake1                 NaN                    NaN\n",
       "1     HG00097     fake2        GAZ:00002641                England\n",
       "2     HG00099     fake3        GAZ:00002640                  Wales\n",
       "3     HG00100     fake4                 NaN                    NaN\n",
       "4     HG00101     fake5                 NaN                    NaN\n",
       "...       ...       ...                 ...                    ...\n",
       "2499  NA21137  fake2500        GAZ:00002638       Northern Ireland\n",
       "2500  NA21141  fake2501                 NaN                    NaN\n",
       "2501  NA21142  fake2502                 NaN                    NaN\n",
       "2502  NA21143  fake2503        GAZ:00002638       Northern Ireland\n",
       "2503  NA21144  fake2504                 NaN                    NaN\n",
       "\n",
       "[2504 rows x 4 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[['id', 'info.eid', 'geographicOrigin.id', 'geographicOrigin.label']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {},
     "inputWidgets": {},
     "nuid": "8c737107-d91a-4667-85d0-2df451693ed1",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>info.eid</th>\n",
       "      <th>pedigrees_disease.ageOfOnset</th>\n",
       "      <th>pedigrees_disease.diseaseCode.id</th>\n",
       "      <th>pedigrees_disease.diseaseCode.label</th>\n",
       "      <th>pedigrees_disease.familyHistory</th>\n",
       "      <th>pedigrees_disease.notes</th>\n",
       "      <th>pedigrees_disease.severity.id</th>\n",
       "      <th>pedigrees_disease.severity.label</th>\n",
       "      <th>pedigrees_disease.stage.id</th>\n",
       "      <th>pedigrees_disease.stage.label</th>\n",
       "      <th>pedigrees_id</th>\n",
       "      <th>pedigrees_members</th>\n",
       "      <th>pedigrees_numSubjects</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HG00096</td>\n",
       "      <td>fake1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HG00097</td>\n",
       "      <td>fake2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HG00099</td>\n",
       "      <td>fake3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HG00100</td>\n",
       "      <td>fake4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HG00101</td>\n",
       "      <td>fake5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2499</th>\n",
       "      <td>NA21137</td>\n",
       "      <td>fake2500</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2500</th>\n",
       "      <td>NA21141</td>\n",
       "      <td>fake2501</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2501</th>\n",
       "      <td>NA21142</td>\n",
       "      <td>fake2502</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2502</th>\n",
       "      <td>NA21143</td>\n",
       "      <td>fake2503</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2503</th>\n",
       "      <td>NA21144</td>\n",
       "      <td>fake2504</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2504 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           id  info.eid  pedigrees_disease.ageOfOnset  \\\n",
       "0     HG00096     fake1                           NaN   \n",
       "1     HG00097     fake2                           NaN   \n",
       "2     HG00099     fake3                           NaN   \n",
       "3     HG00100     fake4                           NaN   \n",
       "4     HG00101     fake5                           NaN   \n",
       "...       ...       ...                           ...   \n",
       "2499  NA21137  fake2500                           NaN   \n",
       "2500  NA21141  fake2501                           NaN   \n",
       "2501  NA21142  fake2502                           NaN   \n",
       "2502  NA21143  fake2503                           NaN   \n",
       "2503  NA21144  fake2504                           NaN   \n",
       "\n",
       "      pedigrees_disease.diseaseCode.id  pedigrees_disease.diseaseCode.label  \\\n",
       "0                                  NaN                                  NaN   \n",
       "1                                  NaN                                  NaN   \n",
       "2                                  NaN                                  NaN   \n",
       "3                                  NaN                                  NaN   \n",
       "4                                  NaN                                  NaN   \n",
       "...                                ...                                  ...   \n",
       "2499                               NaN                                  NaN   \n",
       "2500                               NaN                                  NaN   \n",
       "2501                               NaN                                  NaN   \n",
       "2502                               NaN                                  NaN   \n",
       "2503                               NaN                                  NaN   \n",
       "\n",
       "      pedigrees_disease.familyHistory  pedigrees_disease.notes  \\\n",
       "0                                 NaN                      NaN   \n",
       "1                                 NaN                      NaN   \n",
       "2                                 NaN                      NaN   \n",
       "3                                 NaN                      NaN   \n",
       "4                                 NaN                      NaN   \n",
       "...                               ...                      ...   \n",
       "2499                              NaN                      NaN   \n",
       "2500                              NaN                      NaN   \n",
       "2501                              NaN                      NaN   \n",
       "2502                              NaN                      NaN   \n",
       "2503                              NaN                      NaN   \n",
       "\n",
       "      pedigrees_disease.severity.id  pedigrees_disease.severity.label  \\\n",
       "0                               NaN                               NaN   \n",
       "1                               NaN                               NaN   \n",
       "2                               NaN                               NaN   \n",
       "3                               NaN                               NaN   \n",
       "4                               NaN                               NaN   \n",
       "...                             ...                               ...   \n",
       "2499                            NaN                               NaN   \n",
       "2500                            NaN                               NaN   \n",
       "2501                            NaN                               NaN   \n",
       "2502                            NaN                               NaN   \n",
       "2503                            NaN                               NaN   \n",
       "\n",
       "      pedigrees_disease.stage.id  pedigrees_disease.stage.label  pedigrees_id  \\\n",
       "0                            NaN                            NaN           NaN   \n",
       "1                            NaN                            NaN           NaN   \n",
       "2                            NaN                            NaN           NaN   \n",
       "3                            NaN                            NaN           NaN   \n",
       "4                            NaN                            NaN           NaN   \n",
       "...                          ...                            ...           ...   \n",
       "2499                         NaN                            NaN           NaN   \n",
       "2500                         NaN                            NaN           NaN   \n",
       "2501                         NaN                            NaN           NaN   \n",
       "2502                         NaN                            NaN           NaN   \n",
       "2503                         NaN                            NaN           NaN   \n",
       "\n",
       "      pedigrees_members  pedigrees_numSubjects  \n",
       "0                   NaN                    NaN  \n",
       "1                   NaN                    NaN  \n",
       "2                   NaN                    NaN  \n",
       "3                   NaN                    NaN  \n",
       "4                   NaN                    NaN  \n",
       "...                 ...                    ...  \n",
       "2499                NaN                    NaN  \n",
       "2500                NaN                    NaN  \n",
       "2501                NaN                    NaN  \n",
       "2502                NaN                    NaN  \n",
       "2503                NaN                    NaN  \n",
       "\n",
       "[2504 rows x 14 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[['id', 'info.eid', 'pedigrees_disease.ageOfOnset', 'pedigrees_disease.diseaseCode.id', 'pedigrees_disease.diseaseCode.label', 'pedigrees_disease.familyHistory', 'pedigrees_disease.notes', 'pedigrees_disease.severity.id', 'pedigrees_disease.severity.label', 'pedigrees_disease.stage.id', 'pedigrees_disease.stage.label', 'pedigrees_id', 'pedigrees_members', 'pedigrees_numSubjects']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {},
     "inputWidgets": {},
     "nuid": "f70067ff-a5d9-4d82-94b8-4371d18c029b",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "skipping geographicOrigin for 0 due to id = NaN\n",
      "skipping diseases for 0 due to id = NaN\n",
      "skipping diseases for 1 due to id = NaN\n",
      "skipping diseases for 2 due to id = NaN\n",
      "skipping geographicOrigin for 3 due to id = NaN\n",
      "skipping diseases for 3 due to id = NaN\n",
      "skipping geographicOrigin for 4 due to id = NaN\n",
      "skipping diseases for 4 due to id = NaN\n",
      "skipping ethnicity for 5 due to id = NaN\n",
      "skipping diseases for 5 due to id = NaN\n",
      "skipping geographicOrigin for 6 due to id = NaN\n",
      "skipping diseases for 8 due to id = NaN\n",
      "skipping ethnicity for 12 due to id = NaN\n",
      "skipping diseases for 12 due to id = NaN\n",
      "skipping geographicOrigin for 13 due to id = NaN\n",
      "skipping ethnicity for 14 due to id = NaN\n",
      "skipping geographicOrigin for 15 due to id = NaN\n",
      "skipping ethnicity for 16 due to id = NaN\n",
      "skipping ethnicity for 17 due to id = NaN\n",
      "skipping geographicOrigin for 17 due to id = NaN\n",
      "skipping diseases for 18 due to id = NaN\n",
      "skipping geographicOrigin for 19 due to id = NaN\n",
      "skipping diseases for 21 due to id = NaN\n",
      "skipping ethnicity for 22 due to id = NaN\n",
      "skipping diseases for 22 due to id = NaN\n",
      "skipping geographicOrigin for 23 due to id = NaN\n",
      "skipping geographicOrigin for 27 due to id = NaN\n",
      "skipping geographicOrigin for 30 due to id = NaN\n",
      "skipping diseases for 30 due to id = NaN\n",
      "skipping ethnicity for 31 due to id = NaN\n",
      "skipping geographicOrigin for 31 due to id = NaN\n",
      "skipping geographicOrigin for 32 due to id = NaN\n",
      "skipping geographicOrigin for 34 due to id = NaN\n",
      "skipping geographicOrigin for 35 due to id = NaN\n",
      "skipping geographicOrigin for 36 due to id = NaN\n",
      "skipping geographicOrigin for 38 due to id = NaN\n",
      "skipping diseases for 40 due to id = NaN\n",
      "skipping diseases for 41 due to id = NaN\n",
      "skipping diseases for 47 due to id = NaN\n",
      "skipping geographicOrigin for 48 due to id = NaN\n",
      "skipping geographicOrigin for 50 due to id = NaN\n",
      "skipping geographicOrigin for 51 due to id = NaN\n",
      "skipping geographicOrigin for 52 due to id = NaN\n",
      "skipping ethnicity for 54 due to id = NaN\n",
      "skipping geographicOrigin for 54 due to id = NaN\n",
      "skipping diseases for 55 due to id = NaN\n",
      "skipping diseases for 57 due to id = NaN\n",
      "skipping diseases for 58 due to id = NaN\n",
      "skipping diseases for 63 due to id = NaN\n",
      "skipping geographicOrigin for 64 due to id = NaN\n",
      "skipping diseases for 65 due to id = NaN\n",
      "skipping geographicOrigin for 66 due to id = NaN\n",
      "skipping diseases for 66 due to id = NaN\n",
      "skipping diseases for 68 due to id = NaN\n",
      "skipping diseases for 72 due to id = NaN\n",
      "skipping geographicOrigin for 74 due to id = NaN\n",
      "skipping geographicOrigin for 75 due to id = NaN\n",
      "skipping diseases for 76 due to id = NaN\n",
      "skipping diseases for 78 due to id = NaN\n",
      "skipping geographicOrigin for 79 due to id = NaN\n",
      "skipping geographicOrigin for 81 due to id = NaN\n",
      "skipping geographicOrigin for 82 due to id = NaN\n",
      "skipping diseases for 83 due to id = NaN\n",
      "skipping diseases for 86 due to id = NaN\n",
      "skipping geographicOrigin for 87 due to id = NaN\n",
      "skipping diseases for 87 due to id = NaN\n",
      "skipping geographicOrigin for 88 due to id = NaN\n",
      "skipping geographicOrigin for 89 due to id = NaN\n",
      "skipping geographicOrigin for 90 due to id = NaN\n",
      "skipping ethnicity for 91 due to id = NaN\n",
      "skipping geographicOrigin for 92 due to id = NaN\n",
      "skipping ethnicity for 94 due to id = NaN\n",
      "skipping diseases for 94 due to id = NaN\n",
      "skipping ethnicity for 96 due to id = NaN\n",
      "skipping geographicOrigin for 96 due to id = NaN\n",
      "skipping geographicOrigin for 97 due to id = NaN\n",
      "skipping geographicOrigin for 98 due to id = NaN\n",
      "skipping diseases for 99 due to id = NaN\n",
      "skipping ethnicity for 100 due to id = NaN\n",
      "skipping geographicOrigin for 104 due to id = NaN\n",
      "skipping diseases for 106 due to id = NaN\n",
      "skipping diseases for 107 due to id = NaN\n",
      "skipping geographicOrigin for 108 due to id = NaN\n",
      "skipping diseases for 108 due to id = NaN\n",
      "skipping geographicOrigin for 110 due to id = NaN\n",
      "skipping diseases for 110 due to id = NaN\n",
      "skipping diseases for 112 due to id = NaN\n",
      "skipping geographicOrigin for 115 due to id = NaN\n",
      "skipping ethnicity for 119 due to id = NaN\n",
      "skipping ethnicity for 122 due to id = NaN\n",
      "skipping geographicOrigin for 122 due to id = NaN\n",
      "skipping geographicOrigin for 124 due to id = NaN\n",
      "skipping diseases for 124 due to id = NaN\n",
      "skipping geographicOrigin for 126 due to id = NaN\n",
      "skipping diseases for 129 due to id = NaN\n",
      "skipping diseases for 130 due to id = NaN\n",
      "skipping diseases for 133 due to id = NaN\n",
      "skipping diseases for 136 due to id = NaN\n",
      "skipping geographicOrigin for 137 due to id = NaN\n",
      "skipping geographicOrigin for 141 due to id = NaN\n",
      "skipping diseases for 141 due to id = NaN\n",
      "skipping diseases for 143 due to id = NaN\n",
      "skipping diseases for 144 due to id = NaN\n",
      "skipping diseases for 145 due to id = NaN\n",
      "skipping geographicOrigin for 147 due to id = NaN\n",
      "skipping diseases for 148 due to id = NaN\n",
      "skipping geographicOrigin for 150 due to id = NaN\n",
      "skipping geographicOrigin for 151 due to id = NaN\n",
      "skipping ethnicity for 152 due to id = NaN\n",
      "skipping diseases for 152 due to id = NaN\n",
      "skipping diseases for 153 due to id = NaN\n",
      "skipping geographicOrigin for 154 due to id = NaN\n",
      "skipping geographicOrigin for 155 due to id = NaN\n",
      "skipping diseases for 155 due to id = NaN\n",
      "skipping geographicOrigin for 158 due to id = NaN\n",
      "skipping geographicOrigin for 159 due to id = NaN\n",
      "skipping geographicOrigin for 161 due to id = NaN\n",
      "skipping diseases for 162 due to id = NaN\n",
      "skipping geographicOrigin for 163 due to id = NaN\n",
      "skipping geographicOrigin for 164 due to id = NaN\n",
      "skipping diseases for 164 due to id = NaN\n",
      "skipping geographicOrigin for 165 due to id = NaN\n",
      "skipping diseases for 166 due to id = NaN\n",
      "skipping geographicOrigin for 168 due to id = NaN\n",
      "skipping diseases for 168 due to id = NaN\n",
      "skipping ethnicity for 169 due to id = NaN\n",
      "skipping diseases for 169 due to id = NaN\n",
      "skipping geographicOrigin for 170 due to id = NaN\n",
      "skipping diseases for 170 due to id = NaN\n",
      "skipping geographicOrigin for 172 due to id = NaN\n",
      "skipping geographicOrigin for 173 due to id = NaN\n",
      "skipping geographicOrigin for 174 due to id = NaN\n",
      "skipping diseases for 175 due to id = NaN\n",
      "skipping diseases for 176 due to id = NaN\n",
      "skipping geographicOrigin for 177 due to id = NaN\n",
      "skipping diseases for 177 due to id = NaN\n",
      "skipping geographicOrigin for 180 due to id = NaN\n",
      "skipping geographicOrigin for 181 due to id = NaN\n",
      "skipping diseases for 181 due to id = NaN\n",
      "skipping geographicOrigin for 183 due to id = NaN\n",
      "skipping geographicOrigin for 185 due to id = NaN\n",
      "skipping geographicOrigin for 191 due to id = NaN\n",
      "skipping diseases for 192 due to id = NaN\n",
      "skipping geographicOrigin for 195 due to id = NaN\n",
      "skipping diseases for 196 due to id = NaN\n",
      "skipping geographicOrigin for 201 due to id = NaN\n",
      "skipping geographicOrigin for 203 due to id = NaN\n",
      "skipping diseases for 203 due to id = NaN\n",
      "skipping ethnicity for 205 due to id = NaN\n",
      "skipping geographicOrigin for 205 due to id = NaN\n",
      "skipping diseases for 207 due to id = NaN\n",
      "skipping ethnicity for 208 due to id = NaN\n",
      "skipping geographicOrigin for 208 due to id = NaN\n",
      "skipping geographicOrigin for 209 due to id = NaN\n",
      "skipping diseases for 209 due to id = NaN\n",
      "skipping ethnicity for 210 due to id = NaN\n",
      "skipping geographicOrigin for 210 due to id = NaN\n",
      "skipping geographicOrigin for 211 due to id = NaN\n",
      "skipping diseases for 212 due to id = NaN\n",
      "skipping geographicOrigin for 215 due to id = NaN\n",
      "skipping ethnicity for 217 due to id = NaN\n",
      "skipping geographicOrigin for 219 due to id = NaN\n",
      "skipping diseases for 220 due to id = NaN\n",
      "skipping diseases for 221 due to id = NaN\n",
      "skipping diseases for 222 due to id = NaN\n",
      "skipping diseases for 223 due to id = NaN\n",
      "skipping diseases for 224 due to id = NaN\n",
      "skipping geographicOrigin for 225 due to id = NaN\n",
      "skipping geographicOrigin for 227 due to id = NaN\n",
      "skipping geographicOrigin for 230 due to id = NaN\n",
      "skipping diseases for 230 due to id = NaN\n",
      "skipping geographicOrigin for 231 due to id = NaN\n",
      "skipping geographicOrigin for 233 due to id = NaN\n",
      "skipping diseases for 234 due to id = NaN\n",
      "skipping geographicOrigin for 236 due to id = NaN\n",
      "skipping geographicOrigin for 239 due to id = NaN\n",
      "skipping geographicOrigin for 242 due to id = NaN\n",
      "skipping diseases for 242 due to id = NaN\n",
      "skipping diseases for 243 due to id = NaN\n",
      "skipping diseases for 245 due to id = NaN\n",
      "skipping geographicOrigin for 248 due to id = NaN\n",
      "skipping diseases for 248 due to id = NaN\n",
      "skipping geographicOrigin for 249 due to id = NaN\n",
      "skipping diseases for 249 due to id = NaN\n",
      "skipping geographicOrigin for 251 due to id = NaN\n",
      "skipping diseases for 253 due to id = NaN\n",
      "skipping diseases for 254 due to id = NaN\n",
      "skipping geographicOrigin for 255 due to id = NaN\n",
      "skipping ethnicity for 259 due to id = NaN\n",
      "skipping diseases for 259 due to id = NaN\n",
      "skipping ethnicity for 260 due to id = NaN\n",
      "skipping diseases for 263 due to id = NaN\n",
      "skipping ethnicity for 264 due to id = NaN\n",
      "skipping geographicOrigin for 264 due to id = NaN\n",
      "skipping ethnicity for 265 due to id = NaN\n",
      "skipping geographicOrigin for 265 due to id = NaN\n",
      "skipping geographicOrigin for 266 due to id = NaN\n",
      "skipping diseases for 266 due to id = NaN\n",
      "skipping geographicOrigin for 268 due to id = NaN\n",
      "skipping diseases for 269 due to id = NaN\n",
      "skipping geographicOrigin for 270 due to id = NaN\n",
      "skipping diseases for 270 due to id = NaN\n",
      "skipping diseases for 274 due to id = NaN\n",
      "skipping geographicOrigin for 275 due to id = NaN\n",
      "skipping diseases for 275 due to id = NaN\n",
      "skipping ethnicity for 276 due to id = NaN\n",
      "skipping diseases for 276 due to id = NaN\n",
      "skipping diseases for 278 due to id = NaN\n",
      "skipping diseases for 280 due to id = NaN\n",
      "skipping geographicOrigin for 283 due to id = NaN\n",
      "skipping diseases for 283 due to id = NaN\n",
      "skipping diseases for 284 due to id = NaN\n",
      "skipping diseases for 286 due to id = NaN\n",
      "skipping ethnicity for 289 due to id = NaN\n",
      "skipping diseases for 290 due to id = NaN\n",
      "skipping geographicOrigin for 291 due to id = NaN\n",
      "skipping diseases for 291 due to id = NaN\n",
      "skipping geographicOrigin for 292 due to id = NaN\n",
      "skipping geographicOrigin for 293 due to id = NaN\n",
      "skipping geographicOrigin for 294 due to id = NaN\n",
      "skipping ethnicity for 296 due to id = NaN\n",
      "skipping geographicOrigin for 296 due to id = NaN\n",
      "skipping geographicOrigin for 297 due to id = NaN\n",
      "skipping diseases for 300 due to id = NaN\n",
      "skipping geographicOrigin for 304 due to id = NaN\n",
      "skipping diseases for 307 due to id = NaN\n",
      "skipping geographicOrigin for 308 due to id = NaN\n",
      "skipping diseases for 308 due to id = NaN\n",
      "skipping diseases for 309 due to id = NaN\n",
      "skipping geographicOrigin for 311 due to id = NaN\n",
      "skipping geographicOrigin for 313 due to id = NaN\n",
      "skipping ethnicity for 314 due to id = NaN\n",
      "skipping geographicOrigin for 314 due to id = NaN\n",
      "skipping geographicOrigin for 315 due to id = NaN\n",
      "skipping diseases for 315 due to id = NaN\n",
      "skipping ethnicity for 316 due to id = NaN\n",
      "skipping geographicOrigin for 316 due to id = NaN\n",
      "skipping geographicOrigin for 317 due to id = NaN\n",
      "skipping diseases for 319 due to id = NaN\n",
      "skipping geographicOrigin for 322 due to id = NaN\n",
      "skipping ethnicity for 324 due to id = NaN\n",
      "skipping diseases for 325 due to id = NaN\n",
      "skipping geographicOrigin for 332 due to id = NaN\n",
      "skipping diseases for 332 due to id = NaN\n",
      "skipping geographicOrigin for 334 due to id = NaN\n",
      "skipping diseases for 334 due to id = NaN\n",
      "skipping geographicOrigin for 335 due to id = NaN\n",
      "skipping diseases for 335 due to id = NaN\n",
      "skipping diseases for 338 due to id = NaN\n",
      "skipping geographicOrigin for 339 due to id = NaN\n",
      "skipping geographicOrigin for 340 due to id = NaN\n",
      "skipping diseases for 343 due to id = NaN\n",
      "skipping geographicOrigin for 345 due to id = NaN\n",
      "skipping geographicOrigin for 346 due to id = NaN\n",
      "skipping diseases for 346 due to id = NaN\n",
      "skipping diseases for 349 due to id = NaN\n",
      "skipping ethnicity for 351 due to id = NaN\n",
      "skipping geographicOrigin for 351 due to id = NaN\n",
      "skipping diseases for 351 due to id = NaN\n",
      "skipping geographicOrigin for 353 due to id = NaN\n",
      "skipping diseases for 353 due to id = NaN\n",
      "skipping geographicOrigin for 355 due to id = NaN\n",
      "skipping diseases for 355 due to id = NaN\n",
      "skipping geographicOrigin for 358 due to id = NaN\n",
      "skipping diseases for 358 due to id = NaN\n",
      "skipping diseases for 361 due to id = NaN\n",
      "skipping geographicOrigin for 364 due to id = NaN\n",
      "skipping diseases for 364 due to id = NaN\n",
      "skipping diseases for 365 due to id = NaN\n",
      "skipping diseases for 366 due to id = NaN\n",
      "skipping ethnicity for 367 due to id = NaN\n",
      "skipping geographicOrigin for 368 due to id = NaN\n",
      "skipping ethnicity for 369 due to id = NaN\n",
      "skipping geographicOrigin for 371 due to id = NaN\n",
      "skipping geographicOrigin for 373 due to id = NaN\n",
      "skipping geographicOrigin for 374 due to id = NaN\n",
      "skipping diseases for 375 due to id = NaN\n",
      "skipping geographicOrigin for 376 due to id = NaN\n",
      "skipping diseases for 376 due to id = NaN\n",
      "skipping ethnicity for 378 due to id = NaN\n",
      "skipping geographicOrigin for 379 due to id = NaN\n",
      "skipping geographicOrigin for 381 due to id = NaN\n",
      "skipping diseases for 381 due to id = NaN\n",
      "skipping diseases for 382 due to id = NaN\n",
      "skipping geographicOrigin for 383 due to id = NaN\n",
      "skipping diseases for 386 due to id = NaN\n",
      "skipping geographicOrigin for 387 due to id = NaN\n",
      "skipping diseases for 387 due to id = NaN\n",
      "skipping geographicOrigin for 389 due to id = NaN\n",
      "skipping geographicOrigin for 390 due to id = NaN\n",
      "skipping diseases for 390 due to id = NaN\n",
      "skipping geographicOrigin for 391 due to id = NaN\n",
      "skipping diseases for 391 due to id = NaN\n",
      "skipping diseases for 392 due to id = NaN\n",
      "skipping diseases for 393 due to id = NaN\n",
      "skipping geographicOrigin for 395 due to id = NaN\n",
      "skipping ethnicity for 396 due to id = NaN\n",
      "skipping diseases for 396 due to id = NaN\n",
      "skipping geographicOrigin for 400 due to id = NaN\n",
      "skipping geographicOrigin for 401 due to id = NaN\n",
      "skipping geographicOrigin for 402 due to id = NaN\n",
      "skipping geographicOrigin for 403 due to id = NaN\n",
      "skipping diseases for 403 due to id = NaN\n",
      "skipping ethnicity for 407 due to id = NaN\n",
      "skipping geographicOrigin for 411 due to id = NaN\n",
      "skipping geographicOrigin for 412 due to id = NaN\n",
      "skipping geographicOrigin for 414 due to id = NaN\n",
      "skipping diseases for 414 due to id = NaN\n",
      "skipping geographicOrigin for 418 due to id = NaN\n",
      "skipping geographicOrigin for 420 due to id = NaN\n",
      "skipping geographicOrigin for 423 due to id = NaN\n",
      "skipping ethnicity for 424 due to id = NaN\n",
      "skipping geographicOrigin for 424 due to id = NaN\n",
      "skipping ethnicity for 427 due to id = NaN\n",
      "skipping diseases for 427 due to id = NaN\n",
      "skipping ethnicity for 432 due to id = NaN\n",
      "skipping diseases for 432 due to id = NaN\n",
      "skipping diseases for 434 due to id = NaN\n",
      "skipping geographicOrigin for 436 due to id = NaN\n",
      "skipping geographicOrigin for 439 due to id = NaN\n",
      "skipping geographicOrigin for 440 due to id = NaN\n",
      "skipping ethnicity for 444 due to id = NaN\n",
      "skipping diseases for 444 due to id = NaN\n",
      "skipping geographicOrigin for 445 due to id = NaN\n",
      "skipping diseases for 445 due to id = NaN\n",
      "skipping geographicOrigin for 446 due to id = NaN\n",
      "skipping geographicOrigin for 447 due to id = NaN\n",
      "skipping diseases for 450 due to id = NaN\n",
      "skipping diseases for 451 due to id = NaN\n",
      "skipping geographicOrigin for 452 due to id = NaN\n",
      "skipping diseases for 453 due to id = NaN\n",
      "skipping geographicOrigin for 455 due to id = NaN\n",
      "skipping diseases for 456 due to id = NaN\n",
      "skipping geographicOrigin for 459 due to id = NaN\n",
      "skipping diseases for 460 due to id = NaN\n",
      "skipping geographicOrigin for 461 due to id = NaN\n",
      "skipping ethnicity for 462 due to id = NaN\n",
      "skipping diseases for 462 due to id = NaN\n",
      "skipping geographicOrigin for 464 due to id = NaN\n",
      "skipping diseases for 464 due to id = NaN\n",
      "skipping geographicOrigin for 466 due to id = NaN\n",
      "skipping ethnicity for 468 due to id = NaN\n",
      "skipping geographicOrigin for 471 due to id = NaN\n",
      "skipping diseases for 471 due to id = NaN\n",
      "skipping geographicOrigin for 473 due to id = NaN\n",
      "skipping geographicOrigin for 475 due to id = NaN\n",
      "skipping diseases for 475 due to id = NaN\n",
      "skipping diseases for 476 due to id = NaN\n",
      "skipping diseases for 477 due to id = NaN\n",
      "skipping geographicOrigin for 481 due to id = NaN\n",
      "skipping geographicOrigin for 486 due to id = NaN\n",
      "skipping diseases for 487 due to id = NaN\n",
      "skipping geographicOrigin for 489 due to id = NaN\n",
      "skipping geographicOrigin for 490 due to id = NaN\n",
      "skipping ethnicity for 492 due to id = NaN\n",
      "skipping geographicOrigin for 492 due to id = NaN\n",
      "skipping diseases for 493 due to id = NaN\n",
      "skipping geographicOrigin for 498 due to id = NaN\n",
      "skipping geographicOrigin for 500 due to id = NaN\n",
      "skipping diseases for 502 due to id = NaN\n",
      "skipping geographicOrigin for 503 due to id = NaN\n",
      "skipping ethnicity for 504 due to id = NaN\n",
      "skipping geographicOrigin for 504 due to id = NaN\n",
      "skipping diseases for 504 due to id = NaN\n",
      "skipping ethnicity for 505 due to id = NaN\n",
      "skipping diseases for 505 due to id = NaN\n",
      "skipping diseases for 506 due to id = NaN\n",
      "skipping geographicOrigin for 510 due to id = NaN\n",
      "skipping diseases for 510 due to id = NaN\n",
      "skipping geographicOrigin for 511 due to id = NaN\n",
      "skipping geographicOrigin for 512 due to id = NaN\n",
      "skipping geographicOrigin for 513 due to id = NaN\n",
      "skipping geographicOrigin for 515 due to id = NaN\n",
      "skipping geographicOrigin for 516 due to id = NaN\n",
      "skipping diseases for 518 due to id = NaN\n",
      "skipping geographicOrigin for 519 due to id = NaN\n",
      "skipping diseases for 519 due to id = NaN\n",
      "skipping ethnicity for 523 due to id = NaN\n",
      "skipping diseases for 524 due to id = NaN\n",
      "skipping geographicOrigin for 525 due to id = NaN\n",
      "skipping diseases for 526 due to id = NaN\n",
      "skipping geographicOrigin for 527 due to id = NaN\n",
      "skipping ethnicity for 528 due to id = NaN\n",
      "skipping geographicOrigin for 529 due to id = NaN\n",
      "skipping ethnicity for 530 due to id = NaN\n",
      "skipping diseases for 531 due to id = NaN\n",
      "skipping geographicOrigin for 533 due to id = NaN\n",
      "skipping geographicOrigin for 534 due to id = NaN\n",
      "skipping diseases for 535 due to id = NaN\n",
      "skipping diseases for 536 due to id = NaN\n",
      "skipping geographicOrigin for 537 due to id = NaN\n",
      "skipping diseases for 539 due to id = NaN\n",
      "skipping geographicOrigin for 544 due to id = NaN\n",
      "skipping ethnicity for 545 due to id = NaN\n",
      "skipping geographicOrigin for 545 due to id = NaN\n",
      "skipping diseases for 547 due to id = NaN\n",
      "skipping geographicOrigin for 548 due to id = NaN\n",
      "skipping ethnicity for 549 due to id = NaN\n",
      "skipping diseases for 551 due to id = NaN\n",
      "skipping geographicOrigin for 552 due to id = NaN\n",
      "skipping diseases for 553 due to id = NaN\n",
      "skipping geographicOrigin for 554 due to id = NaN\n",
      "skipping geographicOrigin for 556 due to id = NaN\n",
      "skipping diseases for 556 due to id = NaN\n",
      "skipping geographicOrigin for 557 due to id = NaN\n",
      "skipping diseases for 557 due to id = NaN\n",
      "skipping diseases for 558 due to id = NaN\n",
      "skipping ethnicity for 559 due to id = NaN\n",
      "skipping geographicOrigin for 559 due to id = NaN\n",
      "skipping geographicOrigin for 560 due to id = NaN\n",
      "skipping diseases for 560 due to id = NaN\n",
      "skipping diseases for 561 due to id = NaN\n",
      "skipping ethnicity for 562 due to id = NaN\n",
      "skipping geographicOrigin for 563 due to id = NaN\n",
      "skipping geographicOrigin for 564 due to id = NaN\n",
      "skipping diseases for 564 due to id = NaN\n",
      "skipping ethnicity for 565 due to id = NaN\n",
      "skipping geographicOrigin for 565 due to id = NaN\n",
      "skipping diseases for 570 due to id = NaN\n",
      "skipping geographicOrigin for 573 due to id = NaN\n",
      "skipping diseases for 579 due to id = NaN\n",
      "skipping diseases for 580 due to id = NaN\n",
      "skipping geographicOrigin for 582 due to id = NaN\n",
      "skipping geographicOrigin for 583 due to id = NaN\n",
      "skipping diseases for 584 due to id = NaN\n",
      "skipping geographicOrigin for 585 due to id = NaN\n",
      "skipping diseases for 586 due to id = NaN\n",
      "skipping diseases for 588 due to id = NaN\n",
      "skipping diseases for 589 due to id = NaN\n",
      "skipping diseases for 590 due to id = NaN\n",
      "skipping geographicOrigin for 591 due to id = NaN\n",
      "skipping geographicOrigin for 592 due to id = NaN\n",
      "skipping geographicOrigin for 594 due to id = NaN\n",
      "skipping diseases for 596 due to id = NaN\n",
      "skipping geographicOrigin for 597 due to id = NaN\n",
      "skipping diseases for 597 due to id = NaN\n",
      "skipping geographicOrigin for 598 due to id = NaN\n",
      "skipping diseases for 598 due to id = NaN\n",
      "skipping ethnicity for 600 due to id = NaN\n",
      "skipping diseases for 601 due to id = NaN\n",
      "skipping geographicOrigin for 603 due to id = NaN\n",
      "skipping geographicOrigin for 606 due to id = NaN\n",
      "skipping diseases for 607 due to id = NaN\n",
      "skipping geographicOrigin for 608 due to id = NaN\n",
      "skipping geographicOrigin for 609 due to id = NaN\n",
      "skipping diseases for 610 due to id = NaN\n",
      "skipping geographicOrigin for 611 due to id = NaN\n",
      "skipping diseases for 611 due to id = NaN\n",
      "skipping ethnicity for 612 due to id = NaN\n",
      "skipping diseases for 615 due to id = NaN\n",
      "skipping diseases for 616 due to id = NaN\n",
      "skipping geographicOrigin for 617 due to id = NaN\n",
      "skipping diseases for 618 due to id = NaN\n",
      "skipping geographicOrigin for 620 due to id = NaN\n",
      "skipping geographicOrigin for 623 due to id = NaN\n",
      "skipping diseases for 625 due to id = NaN\n",
      "skipping diseases for 626 due to id = NaN\n",
      "skipping geographicOrigin for 631 due to id = NaN\n",
      "skipping diseases for 632 due to id = NaN\n",
      "skipping diseases for 633 due to id = NaN\n",
      "skipping geographicOrigin for 635 due to id = NaN\n",
      "skipping geographicOrigin for 639 due to id = NaN\n",
      "skipping ethnicity for 640 due to id = NaN\n",
      "skipping diseases for 640 due to id = NaN\n",
      "skipping geographicOrigin for 642 due to id = NaN\n",
      "skipping geographicOrigin for 645 due to id = NaN\n",
      "skipping geographicOrigin for 647 due to id = NaN\n",
      "skipping diseases for 648 due to id = NaN\n",
      "skipping diseases for 649 due to id = NaN\n",
      "skipping geographicOrigin for 650 due to id = NaN\n",
      "skipping diseases for 650 due to id = NaN\n",
      "skipping geographicOrigin for 653 due to id = NaN\n",
      "skipping geographicOrigin for 654 due to id = NaN\n",
      "skipping diseases for 654 due to id = NaN\n",
      "skipping geographicOrigin for 655 due to id = NaN\n",
      "skipping diseases for 655 due to id = NaN\n",
      "skipping ethnicity for 656 due to id = NaN\n",
      "skipping geographicOrigin for 657 due to id = NaN\n",
      "skipping diseases for 657 due to id = NaN\n",
      "skipping geographicOrigin for 658 due to id = NaN\n",
      "skipping geographicOrigin for 659 due to id = NaN\n",
      "skipping geographicOrigin for 660 due to id = NaN\n",
      "skipping geographicOrigin for 662 due to id = NaN\n",
      "skipping diseases for 662 due to id = NaN\n",
      "skipping diseases for 664 due to id = NaN\n",
      "skipping diseases for 665 due to id = NaN\n",
      "skipping geographicOrigin for 666 due to id = NaN\n",
      "skipping geographicOrigin for 667 due to id = NaN\n",
      "skipping geographicOrigin for 670 due to id = NaN\n",
      "skipping diseases for 670 due to id = NaN\n",
      "skipping geographicOrigin for 671 due to id = NaN\n",
      "skipping ethnicity for 674 due to id = NaN\n",
      "skipping geographicOrigin for 674 due to id = NaN\n",
      "skipping geographicOrigin for 675 due to id = NaN\n",
      "skipping geographicOrigin for 676 due to id = NaN\n",
      "skipping diseases for 676 due to id = NaN\n",
      "skipping ethnicity for 678 due to id = NaN\n",
      "skipping diseases for 678 due to id = NaN\n",
      "skipping geographicOrigin for 679 due to id = NaN\n",
      "skipping geographicOrigin for 682 due to id = NaN\n",
      "skipping diseases for 685 due to id = NaN\n",
      "skipping ethnicity for 687 due to id = NaN\n",
      "skipping diseases for 688 due to id = NaN\n",
      "skipping diseases for 689 due to id = NaN\n",
      "skipping geographicOrigin for 691 due to id = NaN\n",
      "skipping diseases for 691 due to id = NaN\n",
      "skipping geographicOrigin for 692 due to id = NaN\n",
      "skipping diseases for 694 due to id = NaN\n",
      "skipping geographicOrigin for 695 due to id = NaN\n",
      "skipping geographicOrigin for 697 due to id = NaN\n",
      "skipping diseases for 699 due to id = NaN\n",
      "skipping geographicOrigin for 700 due to id = NaN\n",
      "skipping geographicOrigin for 701 due to id = NaN\n",
      "skipping diseases for 701 due to id = NaN\n",
      "skipping geographicOrigin for 702 due to id = NaN\n",
      "skipping geographicOrigin for 703 due to id = NaN\n",
      "skipping geographicOrigin for 704 due to id = NaN\n",
      "skipping diseases for 706 due to id = NaN\n",
      "skipping diseases for 707 due to id = NaN\n",
      "skipping diseases for 708 due to id = NaN\n",
      "skipping geographicOrigin for 711 due to id = NaN\n",
      "skipping geographicOrigin for 713 due to id = NaN\n",
      "skipping diseases for 713 due to id = NaN\n",
      "skipping geographicOrigin for 715 due to id = NaN\n",
      "skipping diseases for 716 due to id = NaN\n",
      "skipping geographicOrigin for 717 due to id = NaN\n",
      "skipping geographicOrigin for 720 due to id = NaN\n",
      "skipping ethnicity for 721 due to id = NaN\n",
      "skipping geographicOrigin for 722 due to id = NaN\n",
      "skipping diseases for 722 due to id = NaN\n",
      "skipping diseases for 725 due to id = NaN\n",
      "skipping geographicOrigin for 726 due to id = NaN\n",
      "skipping diseases for 726 due to id = NaN\n",
      "skipping geographicOrigin for 728 due to id = NaN\n",
      "skipping ethnicity for 729 due to id = NaN\n",
      "skipping diseases for 729 due to id = NaN\n",
      "skipping geographicOrigin for 730 due to id = NaN\n",
      "skipping geographicOrigin for 732 due to id = NaN\n",
      "skipping diseases for 732 due to id = NaN\n",
      "skipping diseases for 734 due to id = NaN\n",
      "skipping diseases for 735 due to id = NaN\n",
      "skipping diseases for 738 due to id = NaN\n",
      "skipping diseases for 741 due to id = NaN\n",
      "skipping geographicOrigin for 742 due to id = NaN\n",
      "skipping geographicOrigin for 743 due to id = NaN\n",
      "skipping ethnicity for 744 due to id = NaN\n",
      "skipping geographicOrigin for 744 due to id = NaN\n",
      "skipping diseases for 744 due to id = NaN\n",
      "skipping diseases for 745 due to id = NaN\n",
      "skipping geographicOrigin for 746 due to id = NaN\n",
      "skipping diseases for 746 due to id = NaN\n",
      "skipping geographicOrigin for 747 due to id = NaN\n",
      "skipping ethnicity for 750 due to id = NaN\n",
      "skipping diseases for 750 due to id = NaN\n",
      "skipping geographicOrigin for 752 due to id = NaN\n",
      "skipping diseases for 752 due to id = NaN\n",
      "skipping diseases for 756 due to id = NaN\n",
      "skipping ethnicity for 757 due to id = NaN\n",
      "skipping geographicOrigin for 758 due to id = NaN\n",
      "skipping geographicOrigin for 760 due to id = NaN\n",
      "skipping diseases for 761 due to id = NaN\n",
      "skipping ethnicity for 762 due to id = NaN\n",
      "skipping diseases for 762 due to id = NaN\n",
      "skipping geographicOrigin for 763 due to id = NaN\n",
      "skipping diseases for 766 due to id = NaN\n",
      "skipping diseases for 770 due to id = NaN\n",
      "skipping ethnicity for 772 due to id = NaN\n",
      "skipping geographicOrigin for 772 due to id = NaN\n",
      "skipping geographicOrigin for 773 due to id = NaN\n",
      "skipping geographicOrigin for 774 due to id = NaN\n",
      "skipping diseases for 774 due to id = NaN\n",
      "skipping ethnicity for 775 due to id = NaN\n",
      "skipping diseases for 775 due to id = NaN\n",
      "skipping geographicOrigin for 778 due to id = NaN\n",
      "skipping diseases for 779 due to id = NaN\n",
      "skipping diseases for 780 due to id = NaN\n",
      "skipping geographicOrigin for 781 due to id = NaN\n",
      "skipping diseases for 781 due to id = NaN\n",
      "skipping geographicOrigin for 782 due to id = NaN\n",
      "skipping geographicOrigin for 784 due to id = NaN\n",
      "skipping diseases for 784 due to id = NaN\n",
      "skipping diseases for 786 due to id = NaN\n",
      "skipping geographicOrigin for 789 due to id = NaN\n",
      "skipping geographicOrigin for 790 due to id = NaN\n",
      "skipping diseases for 791 due to id = NaN\n",
      "skipping geographicOrigin for 793 due to id = NaN\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "skipping geographicOrigin for 799 due to id = NaN\n",
      "skipping geographicOrigin for 801 due to id = NaN\n",
      "skipping diseases for 801 due to id = NaN\n",
      "skipping diseases for 802 due to id = NaN\n",
      "skipping geographicOrigin for 805 due to id = NaN\n",
      "skipping diseases for 805 due to id = NaN\n",
      "skipping diseases for 807 due to id = NaN\n",
      "skipping diseases for 808 due to id = NaN\n",
      "skipping diseases for 809 due to id = NaN\n",
      "skipping diseases for 811 due to id = NaN\n",
      "skipping geographicOrigin for 812 due to id = NaN\n",
      "skipping diseases for 812 due to id = NaN\n",
      "skipping geographicOrigin for 813 due to id = NaN\n",
      "skipping diseases for 815 due to id = NaN\n",
      "skipping diseases for 816 due to id = NaN\n",
      "skipping diseases for 817 due to id = NaN\n",
      "skipping geographicOrigin for 820 due to id = NaN\n",
      "skipping diseases for 825 due to id = NaN\n",
      "skipping ethnicity for 828 due to id = NaN\n",
      "skipping geographicOrigin for 828 due to id = NaN\n",
      "skipping geographicOrigin for 829 due to id = NaN\n",
      "skipping geographicOrigin for 830 due to id = NaN\n",
      "skipping diseases for 830 due to id = NaN\n",
      "skipping ethnicity for 831 due to id = NaN\n",
      "skipping diseases for 833 due to id = NaN\n",
      "skipping diseases for 834 due to id = NaN\n",
      "skipping diseases for 835 due to id = NaN\n",
      "skipping diseases for 837 due to id = NaN\n",
      "skipping ethnicity for 838 due to id = NaN\n",
      "skipping geographicOrigin for 839 due to id = NaN\n",
      "skipping diseases for 839 due to id = NaN\n",
      "skipping diseases for 841 due to id = NaN\n",
      "skipping ethnicity for 842 due to id = NaN\n",
      "skipping diseases for 845 due to id = NaN\n",
      "skipping diseases for 848 due to id = NaN\n",
      "skipping diseases for 849 due to id = NaN\n",
      "skipping geographicOrigin for 851 due to id = NaN\n",
      "skipping ethnicity for 853 due to id = NaN\n",
      "skipping diseases for 854 due to id = NaN\n",
      "skipping diseases for 855 due to id = NaN\n",
      "skipping geographicOrigin for 857 due to id = NaN\n",
      "skipping diseases for 857 due to id = NaN\n",
      "skipping geographicOrigin for 858 due to id = NaN\n",
      "skipping geographicOrigin for 861 due to id = NaN\n",
      "skipping ethnicity for 862 due to id = NaN\n",
      "skipping geographicOrigin for 863 due to id = NaN\n",
      "skipping diseases for 863 due to id = NaN\n",
      "skipping diseases for 866 due to id = NaN\n",
      "skipping diseases for 867 due to id = NaN\n",
      "skipping geographicOrigin for 870 due to id = NaN\n",
      "skipping diseases for 870 due to id = NaN\n",
      "skipping geographicOrigin for 871 due to id = NaN\n",
      "skipping diseases for 871 due to id = NaN\n",
      "skipping ethnicity for 872 due to id = NaN\n",
      "skipping geographicOrigin for 873 due to id = NaN\n",
      "skipping geographicOrigin for 874 due to id = NaN\n",
      "skipping diseases for 874 due to id = NaN\n",
      "skipping diseases for 875 due to id = NaN\n",
      "skipping ethnicity for 879 due to id = NaN\n",
      "skipping geographicOrigin for 882 due to id = NaN\n",
      "skipping diseases for 882 due to id = NaN\n",
      "skipping diseases for 883 due to id = NaN\n",
      "skipping diseases for 884 due to id = NaN\n",
      "skipping geographicOrigin for 885 due to id = NaN\n",
      "skipping geographicOrigin for 886 due to id = NaN\n",
      "skipping diseases for 886 due to id = NaN\n",
      "skipping geographicOrigin for 887 due to id = NaN\n",
      "skipping geographicOrigin for 889 due to id = NaN\n",
      "skipping diseases for 889 due to id = NaN\n",
      "skipping geographicOrigin for 890 due to id = NaN\n",
      "skipping ethnicity for 891 due to id = NaN\n",
      "skipping diseases for 891 due to id = NaN\n",
      "skipping geographicOrigin for 893 due to id = NaN\n",
      "skipping diseases for 894 due to id = NaN\n",
      "skipping ethnicity for 895 due to id = NaN\n",
      "skipping geographicOrigin for 895 due to id = NaN\n",
      "skipping diseases for 895 due to id = NaN\n",
      "skipping geographicOrigin for 896 due to id = NaN\n",
      "skipping ethnicity for 898 due to id = NaN\n",
      "skipping geographicOrigin for 898 due to id = NaN\n",
      "skipping diseases for 900 due to id = NaN\n",
      "skipping ethnicity for 901 due to id = NaN\n",
      "skipping diseases for 901 due to id = NaN\n",
      "skipping geographicOrigin for 902 due to id = NaN\n",
      "skipping diseases for 903 due to id = NaN\n",
      "skipping geographicOrigin for 904 due to id = NaN\n",
      "skipping diseases for 905 due to id = NaN\n",
      "skipping diseases for 906 due to id = NaN\n",
      "skipping ethnicity for 909 due to id = NaN\n",
      "skipping diseases for 910 due to id = NaN\n",
      "skipping diseases for 911 due to id = NaN\n",
      "skipping geographicOrigin for 912 due to id = NaN\n",
      "skipping diseases for 912 due to id = NaN\n",
      "skipping geographicOrigin for 913 due to id = NaN\n",
      "skipping diseases for 915 due to id = NaN\n",
      "skipping diseases for 916 due to id = NaN\n",
      "skipping geographicOrigin for 918 due to id = NaN\n",
      "skipping diseases for 918 due to id = NaN\n",
      "skipping geographicOrigin for 919 due to id = NaN\n",
      "skipping diseases for 919 due to id = NaN\n",
      "skipping diseases for 920 due to id = NaN\n",
      "skipping geographicOrigin for 922 due to id = NaN\n",
      "skipping diseases for 923 due to id = NaN\n",
      "skipping diseases for 925 due to id = NaN\n",
      "skipping diseases for 927 due to id = NaN\n",
      "skipping geographicOrigin for 928 due to id = NaN\n",
      "skipping diseases for 930 due to id = NaN\n",
      "skipping ethnicity for 932 due to id = NaN\n",
      "skipping geographicOrigin for 932 due to id = NaN\n",
      "skipping geographicOrigin for 934 due to id = NaN\n",
      "skipping diseases for 934 due to id = NaN\n",
      "skipping geographicOrigin for 935 due to id = NaN\n",
      "skipping ethnicity for 936 due to id = NaN\n",
      "skipping diseases for 936 due to id = NaN\n",
      "skipping geographicOrigin for 937 due to id = NaN\n",
      "skipping ethnicity for 939 due to id = NaN\n",
      "skipping geographicOrigin for 939 due to id = NaN\n",
      "skipping ethnicity for 940 due to id = NaN\n",
      "skipping geographicOrigin for 940 due to id = NaN\n",
      "skipping diseases for 941 due to id = NaN\n",
      "skipping geographicOrigin for 943 due to id = NaN\n",
      "skipping diseases for 943 due to id = NaN\n",
      "skipping geographicOrigin for 945 due to id = NaN\n",
      "skipping geographicOrigin for 946 due to id = NaN\n",
      "skipping geographicOrigin for 948 due to id = NaN\n",
      "skipping diseases for 948 due to id = NaN\n",
      "skipping geographicOrigin for 949 due to id = NaN\n",
      "skipping diseases for 949 due to id = NaN\n",
      "skipping geographicOrigin for 951 due to id = NaN\n",
      "skipping diseases for 952 due to id = NaN\n",
      "skipping diseases for 953 due to id = NaN\n",
      "skipping diseases for 955 due to id = NaN\n",
      "skipping geographicOrigin for 956 due to id = NaN\n",
      "skipping geographicOrigin for 957 due to id = NaN\n",
      "skipping ethnicity for 959 due to id = NaN\n",
      "skipping ethnicity for 960 due to id = NaN\n",
      "skipping diseases for 960 due to id = NaN\n",
      "skipping diseases for 961 due to id = NaN\n",
      "skipping geographicOrigin for 964 due to id = NaN\n",
      "skipping diseases for 964 due to id = NaN\n",
      "skipping geographicOrigin for 967 due to id = NaN\n",
      "skipping geographicOrigin for 970 due to id = NaN\n",
      "skipping ethnicity for 971 due to id = NaN\n",
      "skipping diseases for 972 due to id = NaN\n",
      "skipping diseases for 975 due to id = NaN\n",
      "skipping diseases for 976 due to id = NaN\n",
      "skipping diseases for 977 due to id = NaN\n",
      "skipping geographicOrigin for 978 due to id = NaN\n",
      "skipping geographicOrigin for 979 due to id = NaN\n",
      "skipping geographicOrigin for 980 due to id = NaN\n",
      "skipping geographicOrigin for 983 due to id = NaN\n",
      "skipping geographicOrigin for 984 due to id = NaN\n",
      "skipping geographicOrigin for 989 due to id = NaN\n",
      "skipping geographicOrigin for 990 due to id = NaN\n",
      "skipping diseases for 991 due to id = NaN\n",
      "skipping diseases for 992 due to id = NaN\n",
      "skipping diseases for 993 due to id = NaN\n",
      "skipping diseases for 994 due to id = NaN\n",
      "skipping geographicOrigin for 995 due to id = NaN\n",
      "skipping diseases for 995 due to id = NaN\n",
      "skipping geographicOrigin for 996 due to id = NaN\n",
      "skipping diseases for 996 due to id = NaN\n",
      "skipping geographicOrigin for 997 due to id = NaN\n",
      "skipping ethnicity for 998 due to id = NaN\n",
      "skipping geographicOrigin for 998 due to id = NaN\n",
      "skipping geographicOrigin for 999 due to id = NaN\n",
      "skipping geographicOrigin for 1000 due to id = NaN\n",
      "skipping diseases for 1000 due to id = NaN\n",
      "skipping diseases for 1001 due to id = NaN\n",
      "skipping diseases for 1004 due to id = NaN\n",
      "skipping ethnicity for 1005 due to id = NaN\n",
      "skipping geographicOrigin for 1005 due to id = NaN\n",
      "skipping diseases for 1006 due to id = NaN\n",
      "skipping geographicOrigin for 1008 due to id = NaN\n",
      "skipping diseases for 1008 due to id = NaN\n",
      "skipping diseases for 1009 due to id = NaN\n",
      "skipping geographicOrigin for 1011 due to id = NaN\n",
      "skipping diseases for 1011 due to id = NaN\n",
      "skipping geographicOrigin for 1015 due to id = NaN\n",
      "skipping geographicOrigin for 1016 due to id = NaN\n",
      "skipping geographicOrigin for 1017 due to id = NaN\n",
      "skipping diseases for 1019 due to id = NaN\n",
      "skipping ethnicity for 1020 due to id = NaN\n",
      "skipping ethnicity for 1021 due to id = NaN\n",
      "skipping geographicOrigin for 1021 due to id = NaN\n",
      "skipping geographicOrigin for 1023 due to id = NaN\n",
      "skipping geographicOrigin for 1024 due to id = NaN\n",
      "skipping geographicOrigin for 1025 due to id = NaN\n",
      "skipping geographicOrigin for 1026 due to id = NaN\n",
      "skipping geographicOrigin for 1027 due to id = NaN\n",
      "skipping geographicOrigin for 1028 due to id = NaN\n",
      "skipping diseases for 1028 due to id = NaN\n",
      "skipping ethnicity for 1030 due to id = NaN\n",
      "skipping geographicOrigin for 1030 due to id = NaN\n",
      "skipping geographicOrigin for 1031 due to id = NaN\n",
      "skipping geographicOrigin for 1032 due to id = NaN\n",
      "skipping diseases for 1032 due to id = NaN\n",
      "skipping ethnicity for 1034 due to id = NaN\n",
      "skipping diseases for 1036 due to id = NaN\n",
      "skipping geographicOrigin for 1037 due to id = NaN\n",
      "skipping diseases for 1038 due to id = NaN\n",
      "skipping geographicOrigin for 1040 due to id = NaN\n",
      "skipping diseases for 1041 due to id = NaN\n",
      "skipping diseases for 1042 due to id = NaN\n",
      "skipping ethnicity for 1043 due to id = NaN\n",
      "skipping diseases for 1043 due to id = NaN\n",
      "skipping geographicOrigin for 1044 due to id = NaN\n",
      "skipping diseases for 1044 due to id = NaN\n",
      "skipping diseases for 1045 due to id = NaN\n",
      "skipping geographicOrigin for 1046 due to id = NaN\n",
      "skipping geographicOrigin for 1048 due to id = NaN\n",
      "skipping diseases for 1048 due to id = NaN\n",
      "skipping geographicOrigin for 1050 due to id = NaN\n",
      "skipping geographicOrigin for 1051 due to id = NaN\n",
      "skipping diseases for 1052 due to id = NaN\n",
      "skipping diseases for 1053 due to id = NaN\n",
      "skipping diseases for 1055 due to id = NaN\n",
      "skipping geographicOrigin for 1056 due to id = NaN\n",
      "skipping diseases for 1056 due to id = NaN\n",
      "skipping geographicOrigin for 1057 due to id = NaN\n",
      "skipping diseases for 1058 due to id = NaN\n",
      "skipping diseases for 1059 due to id = NaN\n",
      "skipping ethnicity for 1061 due to id = NaN\n",
      "skipping geographicOrigin for 1061 due to id = NaN\n",
      "skipping diseases for 1061 due to id = NaN\n",
      "skipping diseases for 1062 due to id = NaN\n",
      "skipping diseases for 1064 due to id = NaN\n",
      "skipping geographicOrigin for 1065 due to id = NaN\n",
      "skipping diseases for 1065 due to id = NaN\n",
      "skipping geographicOrigin for 1068 due to id = NaN\n",
      "skipping geographicOrigin for 1069 due to id = NaN\n",
      "skipping diseases for 1075 due to id = NaN\n",
      "skipping geographicOrigin for 1077 due to id = NaN\n",
      "skipping diseases for 1077 due to id = NaN\n",
      "skipping geographicOrigin for 1078 due to id = NaN\n",
      "skipping diseases for 1078 due to id = NaN\n",
      "skipping diseases for 1080 due to id = NaN\n",
      "skipping ethnicity for 1082 due to id = NaN\n",
      "skipping geographicOrigin for 1084 due to id = NaN\n",
      "skipping diseases for 1085 due to id = NaN\n",
      "skipping diseases for 1089 due to id = NaN\n",
      "skipping geographicOrigin for 1090 due to id = NaN\n",
      "skipping diseases for 1090 due to id = NaN\n",
      "skipping geographicOrigin for 1091 due to id = NaN\n",
      "skipping geographicOrigin for 1092 due to id = NaN\n",
      "skipping geographicOrigin for 1093 due to id = NaN\n",
      "skipping diseases for 1093 due to id = NaN\n",
      "skipping diseases for 1095 due to id = NaN\n",
      "skipping diseases for 1096 due to id = NaN\n",
      "skipping geographicOrigin for 1097 due to id = NaN\n",
      "skipping geographicOrigin for 1101 due to id = NaN\n",
      "skipping geographicOrigin for 1102 due to id = NaN\n",
      "skipping diseases for 1102 due to id = NaN\n",
      "skipping ethnicity for 1105 due to id = NaN\n",
      "skipping geographicOrigin for 1105 due to id = NaN\n",
      "skipping geographicOrigin for 1107 due to id = NaN\n",
      "skipping geographicOrigin for 1109 due to id = NaN\n",
      "skipping geographicOrigin for 1111 due to id = NaN\n",
      "skipping diseases for 1111 due to id = NaN\n",
      "skipping geographicOrigin for 1114 due to id = NaN\n",
      "skipping ethnicity for 1116 due to id = NaN\n",
      "skipping diseases for 1117 due to id = NaN\n",
      "skipping ethnicity for 1119 due to id = NaN\n",
      "skipping diseases for 1121 due to id = NaN\n",
      "skipping geographicOrigin for 1122 due to id = NaN\n",
      "skipping geographicOrigin for 1126 due to id = NaN\n",
      "skipping diseases for 1128 due to id = NaN\n",
      "skipping geographicOrigin for 1129 due to id = NaN\n",
      "skipping diseases for 1130 due to id = NaN\n",
      "skipping diseases for 1131 due to id = NaN\n",
      "skipping ethnicity for 1132 due to id = NaN\n",
      "skipping diseases for 1132 due to id = NaN\n",
      "skipping diseases for 1133 due to id = NaN\n",
      "skipping geographicOrigin for 1134 due to id = NaN\n",
      "skipping geographicOrigin for 1135 due to id = NaN\n",
      "skipping geographicOrigin for 1137 due to id = NaN\n",
      "skipping diseases for 1138 due to id = NaN\n",
      "skipping geographicOrigin for 1139 due to id = NaN\n",
      "skipping diseases for 1139 due to id = NaN\n",
      "skipping diseases for 1140 due to id = NaN\n",
      "skipping diseases for 1142 due to id = NaN\n",
      "skipping diseases for 1147 due to id = NaN\n",
      "skipping ethnicity for 1148 due to id = NaN\n",
      "skipping diseases for 1149 due to id = NaN\n",
      "skipping ethnicity for 1151 due to id = NaN\n",
      "skipping diseases for 1151 due to id = NaN\n",
      "skipping ethnicity for 1153 due to id = NaN\n",
      "skipping geographicOrigin for 1153 due to id = NaN\n",
      "skipping diseases for 1154 due to id = NaN\n",
      "skipping geographicOrigin for 1155 due to id = NaN\n",
      "skipping diseases for 1156 due to id = NaN\n",
      "skipping geographicOrigin for 1157 due to id = NaN\n",
      "skipping geographicOrigin for 1164 due to id = NaN\n",
      "skipping diseases for 1164 due to id = NaN\n",
      "skipping geographicOrigin for 1166 due to id = NaN\n",
      "skipping diseases for 1166 due to id = NaN\n",
      "skipping ethnicity for 1169 due to id = NaN\n",
      "skipping geographicOrigin for 1170 due to id = NaN\n",
      "skipping diseases for 1171 due to id = NaN\n",
      "skipping diseases for 1173 due to id = NaN\n",
      "skipping diseases for 1176 due to id = NaN\n",
      "skipping geographicOrigin for 1178 due to id = NaN\n",
      "skipping diseases for 1178 due to id = NaN\n",
      "skipping ethnicity for 1180 due to id = NaN\n",
      "skipping diseases for 1180 due to id = NaN\n",
      "skipping ethnicity for 1181 due to id = NaN\n",
      "skipping geographicOrigin for 1181 due to id = NaN\n",
      "skipping geographicOrigin for 1182 due to id = NaN\n",
      "skipping geographicOrigin for 1185 due to id = NaN\n",
      "skipping diseases for 1187 due to id = NaN\n",
      "skipping geographicOrigin for 1188 due to id = NaN\n",
      "skipping geographicOrigin for 1189 due to id = NaN\n",
      "skipping geographicOrigin for 1190 due to id = NaN\n",
      "skipping diseases for 1191 due to id = NaN\n",
      "skipping geographicOrigin for 1192 due to id = NaN\n",
      "skipping geographicOrigin for 1194 due to id = NaN\n",
      "skipping diseases for 1194 due to id = NaN\n",
      "skipping geographicOrigin for 1195 due to id = NaN\n",
      "skipping geographicOrigin for 1196 due to id = NaN\n",
      "skipping geographicOrigin for 1198 due to id = NaN\n",
      "skipping diseases for 1200 due to id = NaN\n",
      "skipping geographicOrigin for 1202 due to id = NaN\n",
      "skipping geographicOrigin for 1206 due to id = NaN\n",
      "skipping diseases for 1206 due to id = NaN\n",
      "skipping diseases for 1207 due to id = NaN\n",
      "skipping geographicOrigin for 1209 due to id = NaN\n",
      "skipping diseases for 1209 due to id = NaN\n",
      "skipping diseases for 1210 due to id = NaN\n",
      "skipping geographicOrigin for 1211 due to id = NaN\n",
      "skipping diseases for 1213 due to id = NaN\n",
      "skipping geographicOrigin for 1214 due to id = NaN\n",
      "skipping diseases for 1214 due to id = NaN\n",
      "skipping diseases for 1215 due to id = NaN\n",
      "skipping ethnicity for 1216 due to id = NaN\n",
      "skipping geographicOrigin for 1216 due to id = NaN\n",
      "skipping diseases for 1216 due to id = NaN\n",
      "skipping ethnicity for 1218 due to id = NaN\n",
      "skipping ethnicity for 1219 due to id = NaN\n",
      "skipping diseases for 1220 due to id = NaN\n",
      "skipping geographicOrigin for 1222 due to id = NaN\n",
      "skipping diseases for 1223 due to id = NaN\n",
      "skipping geographicOrigin for 1224 due to id = NaN\n",
      "skipping diseases for 1225 due to id = NaN\n",
      "skipping geographicOrigin for 1226 due to id = NaN\n",
      "skipping geographicOrigin for 1227 due to id = NaN\n",
      "skipping diseases for 1229 due to id = NaN\n",
      "skipping geographicOrigin for 1230 due to id = NaN\n",
      "skipping diseases for 1231 due to id = NaN\n",
      "skipping diseases for 1232 due to id = NaN\n",
      "skipping diseases for 1233 due to id = NaN\n",
      "skipping geographicOrigin for 1234 due to id = NaN\n",
      "skipping geographicOrigin for 1235 due to id = NaN\n",
      "skipping geographicOrigin for 1236 due to id = NaN\n",
      "skipping geographicOrigin for 1237 due to id = NaN\n",
      "skipping diseases for 1237 due to id = NaN\n",
      "skipping geographicOrigin for 1238 due to id = NaN\n",
      "skipping geographicOrigin for 1239 due to id = NaN\n",
      "skipping geographicOrigin for 1240 due to id = NaN\n",
      "skipping diseases for 1242 due to id = NaN\n",
      "skipping diseases for 1243 due to id = NaN\n",
      "skipping diseases for 1245 due to id = NaN\n",
      "skipping geographicOrigin for 1247 due to id = NaN\n",
      "skipping geographicOrigin for 1249 due to id = NaN\n",
      "skipping diseases for 1249 due to id = NaN\n",
      "skipping diseases for 1251 due to id = NaN\n",
      "skipping geographicOrigin for 1252 due to id = NaN\n",
      "skipping geographicOrigin for 1253 due to id = NaN\n",
      "skipping geographicOrigin for 1254 due to id = NaN\n",
      "skipping diseases for 1254 due to id = NaN\n",
      "skipping geographicOrigin for 1255 due to id = NaN\n",
      "skipping ethnicity for 1257 due to id = NaN\n",
      "skipping geographicOrigin for 1257 due to id = NaN\n",
      "skipping diseases for 1258 due to id = NaN\n",
      "skipping geographicOrigin for 1260 due to id = NaN\n",
      "skipping ethnicity for 1262 due to id = NaN\n",
      "skipping diseases for 1262 due to id = NaN\n",
      "skipping geographicOrigin for 1264 due to id = NaN\n",
      "skipping geographicOrigin for 1265 due to id = NaN\n",
      "skipping diseases for 1265 due to id = NaN\n",
      "skipping diseases for 1266 due to id = NaN\n",
      "skipping geographicOrigin for 1267 due to id = NaN\n",
      "skipping geographicOrigin for 1268 due to id = NaN\n",
      "skipping diseases for 1268 due to id = NaN\n",
      "skipping diseases for 1270 due to id = NaN\n",
      "skipping geographicOrigin for 1272 due to id = NaN\n",
      "skipping diseases for 1272 due to id = NaN\n",
      "skipping diseases for 1274 due to id = NaN\n",
      "skipping diseases for 1275 due to id = NaN\n",
      "skipping geographicOrigin for 1276 due to id = NaN\n",
      "skipping geographicOrigin for 1277 due to id = NaN\n",
      "skipping diseases for 1280 due to id = NaN\n",
      "skipping diseases for 1281 due to id = NaN\n",
      "skipping diseases for 1285 due to id = NaN\n",
      "skipping geographicOrigin for 1288 due to id = NaN\n",
      "skipping diseases for 1289 due to id = NaN\n",
      "skipping geographicOrigin for 1290 due to id = NaN\n",
      "skipping geographicOrigin for 1293 due to id = NaN\n",
      "skipping diseases for 1293 due to id = NaN\n",
      "skipping diseases for 1294 due to id = NaN\n",
      "skipping ethnicity for 1296 due to id = NaN\n",
      "skipping geographicOrigin for 1296 due to id = NaN\n",
      "skipping geographicOrigin for 1298 due to id = NaN\n",
      "skipping geographicOrigin for 1299 due to id = NaN\n",
      "skipping ethnicity for 1301 due to id = NaN\n",
      "skipping geographicOrigin for 1303 due to id = NaN\n",
      "skipping geographicOrigin for 1305 due to id = NaN\n",
      "skipping diseases for 1305 due to id = NaN\n",
      "skipping geographicOrigin for 1308 due to id = NaN\n",
      "skipping geographicOrigin for 1312 due to id = NaN\n",
      "skipping diseases for 1313 due to id = NaN\n",
      "skipping diseases for 1314 due to id = NaN\n",
      "skipping geographicOrigin for 1315 due to id = NaN\n",
      "skipping diseases for 1316 due to id = NaN\n",
      "skipping diseases for 1320 due to id = NaN\n",
      "skipping geographicOrigin for 1321 due to id = NaN\n",
      "skipping geographicOrigin for 1322 due to id = NaN\n",
      "skipping diseases for 1325 due to id = NaN\n",
      "skipping diseases for 1328 due to id = NaN\n",
      "skipping diseases for 1329 due to id = NaN\n",
      "skipping geographicOrigin for 1330 due to id = NaN\n",
      "skipping geographicOrigin for 1331 due to id = NaN\n",
      "skipping geographicOrigin for 1332 due to id = NaN\n",
      "skipping diseases for 1332 due to id = NaN\n",
      "skipping diseases for 1333 due to id = NaN\n",
      "skipping geographicOrigin for 1335 due to id = NaN\n",
      "skipping diseases for 1335 due to id = NaN\n",
      "skipping geographicOrigin for 1336 due to id = NaN\n",
      "skipping diseases for 1337 due to id = NaN\n",
      "skipping diseases for 1338 due to id = NaN\n",
      "skipping diseases for 1339 due to id = NaN\n",
      "skipping ethnicity for 1341 due to id = NaN\n",
      "skipping diseases for 1341 due to id = NaN\n",
      "skipping geographicOrigin for 1343 due to id = NaN\n",
      "skipping diseases for 1343 due to id = NaN\n",
      "skipping geographicOrigin for 1344 due to id = NaN\n",
      "skipping geographicOrigin for 1345 due to id = NaN\n",
      "skipping diseases for 1350 due to id = NaN\n",
      "skipping diseases for 1354 due to id = NaN\n",
      "skipping diseases for 1356 due to id = NaN\n",
      "skipping diseases for 1358 due to id = NaN\n",
      "skipping geographicOrigin for 1360 due to id = NaN\n",
      "skipping diseases for 1360 due to id = NaN\n",
      "skipping geographicOrigin for 1361 due to id = NaN\n",
      "skipping diseases for 1362 due to id = NaN\n",
      "skipping geographicOrigin for 1365 due to id = NaN\n",
      "skipping geographicOrigin for 1367 due to id = NaN\n",
      "skipping diseases for 1367 due to id = NaN\n",
      "skipping diseases for 1370 due to id = NaN\n",
      "skipping ethnicity for 1372 due to id = NaN\n",
      "skipping geographicOrigin for 1372 due to id = NaN\n",
      "skipping geographicOrigin for 1373 due to id = NaN\n",
      "skipping geographicOrigin for 1376 due to id = NaN\n",
      "skipping geographicOrigin for 1377 due to id = NaN\n",
      "skipping geographicOrigin for 1378 due to id = NaN\n",
      "skipping diseases for 1379 due to id = NaN\n",
      "skipping geographicOrigin for 1380 due to id = NaN\n",
      "skipping ethnicity for 1381 due to id = NaN\n",
      "skipping geographicOrigin for 1381 due to id = NaN\n",
      "skipping geographicOrigin for 1382 due to id = NaN\n",
      "skipping geographicOrigin for 1385 due to id = NaN\n",
      "skipping geographicOrigin for 1386 due to id = NaN\n",
      "skipping diseases for 1387 due to id = NaN\n",
      "skipping diseases for 1389 due to id = NaN\n",
      "skipping diseases for 1390 due to id = NaN\n",
      "skipping diseases for 1393 due to id = NaN\n",
      "skipping diseases for 1394 due to id = NaN\n",
      "skipping ethnicity for 1396 due to id = NaN\n",
      "skipping geographicOrigin for 1397 due to id = NaN\n",
      "skipping geographicOrigin for 1399 due to id = NaN\n",
      "skipping diseases for 1400 due to id = NaN\n",
      "skipping diseases for 1401 due to id = NaN\n",
      "skipping geographicOrigin for 1402 due to id = NaN\n",
      "skipping ethnicity for 1403 due to id = NaN\n",
      "skipping geographicOrigin for 1403 due to id = NaN\n",
      "skipping diseases for 1404 due to id = NaN\n",
      "skipping geographicOrigin for 1405 due to id = NaN\n",
      "skipping ethnicity for 1409 due to id = NaN\n",
      "skipping diseases for 1410 due to id = NaN\n",
      "skipping geographicOrigin for 1411 due to id = NaN\n",
      "skipping diseases for 1412 due to id = NaN\n",
      "skipping geographicOrigin for 1413 due to id = NaN\n",
      "skipping diseases for 1414 due to id = NaN\n",
      "skipping geographicOrigin for 1415 due to id = NaN\n",
      "skipping geographicOrigin for 1416 due to id = NaN\n",
      "skipping geographicOrigin for 1417 due to id = NaN\n",
      "skipping geographicOrigin for 1418 due to id = NaN\n",
      "skipping diseases for 1419 due to id = NaN\n",
      "skipping ethnicity for 1421 due to id = NaN\n",
      "skipping diseases for 1422 due to id = NaN\n",
      "skipping geographicOrigin for 1424 due to id = NaN\n",
      "skipping geographicOrigin for 1429 due to id = NaN\n",
      "skipping geographicOrigin for 1430 due to id = NaN\n",
      "skipping diseases for 1431 due to id = NaN\n",
      "skipping geographicOrigin for 1432 due to id = NaN\n",
      "skipping geographicOrigin for 1436 due to id = NaN\n",
      "skipping diseases for 1436 due to id = NaN\n",
      "skipping diseases for 1438 due to id = NaN\n",
      "skipping diseases for 1439 due to id = NaN\n",
      "skipping diseases for 1440 due to id = NaN\n",
      "skipping diseases for 1441 due to id = NaN\n",
      "skipping geographicOrigin for 1442 due to id = NaN\n",
      "skipping diseases for 1442 due to id = NaN\n",
      "skipping geographicOrigin for 1443 due to id = NaN\n",
      "skipping ethnicity for 1444 due to id = NaN\n",
      "skipping geographicOrigin for 1444 due to id = NaN\n",
      "skipping diseases for 1444 due to id = NaN\n",
      "skipping geographicOrigin for 1445 due to id = NaN\n",
      "skipping diseases for 1445 due to id = NaN\n",
      "skipping diseases for 1447 due to id = NaN\n",
      "skipping diseases for 1449 due to id = NaN\n",
      "skipping geographicOrigin for 1450 due to id = NaN\n",
      "skipping diseases for 1450 due to id = NaN\n",
      "skipping geographicOrigin for 1452 due to id = NaN\n",
      "skipping diseases for 1452 due to id = NaN\n",
      "skipping diseases for 1453 due to id = NaN\n",
      "skipping geographicOrigin for 1454 due to id = NaN\n",
      "skipping diseases for 1458 due to id = NaN\n",
      "skipping ethnicity for 1459 due to id = NaN\n",
      "skipping geographicOrigin for 1459 due to id = NaN\n",
      "skipping geographicOrigin for 1461 due to id = NaN\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "skipping diseases for 1464 due to id = NaN\n",
      "skipping geographicOrigin for 1465 due to id = NaN\n",
      "skipping geographicOrigin for 1467 due to id = NaN\n",
      "skipping geographicOrigin for 1468 due to id = NaN\n",
      "skipping geographicOrigin for 1470 due to id = NaN\n",
      "skipping diseases for 1470 due to id = NaN\n",
      "skipping diseases for 1474 due to id = NaN\n",
      "skipping diseases for 1475 due to id = NaN\n",
      "skipping geographicOrigin for 1478 due to id = NaN\n",
      "skipping diseases for 1478 due to id = NaN\n",
      "skipping ethnicity for 1479 due to id = NaN\n",
      "skipping diseases for 1479 due to id = NaN\n",
      "skipping diseases for 1480 due to id = NaN\n",
      "skipping diseases for 1483 due to id = NaN\n",
      "skipping ethnicity for 1487 due to id = NaN\n",
      "skipping geographicOrigin for 1487 due to id = NaN\n",
      "skipping geographicOrigin for 1488 due to id = NaN\n",
      "skipping diseases for 1488 due to id = NaN\n",
      "skipping diseases for 1490 due to id = NaN\n",
      "skipping geographicOrigin for 1491 due to id = NaN\n",
      "skipping geographicOrigin for 1493 due to id = NaN\n",
      "skipping geographicOrigin for 1495 due to id = NaN\n",
      "skipping diseases for 1495 due to id = NaN\n",
      "skipping geographicOrigin for 1496 due to id = NaN\n",
      "skipping geographicOrigin for 1497 due to id = NaN\n",
      "skipping diseases for 1498 due to id = NaN\n",
      "skipping geographicOrigin for 1501 due to id = NaN\n",
      "skipping diseases for 1503 due to id = NaN\n",
      "skipping diseases for 1505 due to id = NaN\n",
      "skipping geographicOrigin for 1506 due to id = NaN\n",
      "skipping diseases for 1506 due to id = NaN\n",
      "skipping diseases for 1508 due to id = NaN\n",
      "skipping diseases for 1510 due to id = NaN\n",
      "skipping diseases for 1511 due to id = NaN\n",
      "skipping diseases for 1512 due to id = NaN\n",
      "skipping geographicOrigin for 1513 due to id = NaN\n",
      "skipping geographicOrigin for 1514 due to id = NaN\n",
      "skipping diseases for 1514 due to id = NaN\n",
      "skipping geographicOrigin for 1515 due to id = NaN\n",
      "skipping diseases for 1518 due to id = NaN\n",
      "skipping diseases for 1519 due to id = NaN\n",
      "skipping ethnicity for 1520 due to id = NaN\n",
      "skipping geographicOrigin for 1521 due to id = NaN\n",
      "skipping geographicOrigin for 1524 due to id = NaN\n",
      "skipping geographicOrigin for 1526 due to id = NaN\n",
      "skipping diseases for 1527 due to id = NaN\n",
      "skipping diseases for 1528 due to id = NaN\n",
      "skipping diseases for 1531 due to id = NaN\n",
      "skipping ethnicity for 1532 due to id = NaN\n",
      "skipping geographicOrigin for 1532 due to id = NaN\n",
      "skipping geographicOrigin for 1533 due to id = NaN\n",
      "skipping diseases for 1533 due to id = NaN\n",
      "skipping geographicOrigin for 1534 due to id = NaN\n",
      "skipping diseases for 1534 due to id = NaN\n",
      "skipping diseases for 1535 due to id = NaN\n",
      "skipping geographicOrigin for 1536 due to id = NaN\n",
      "skipping diseases for 1536 due to id = NaN\n",
      "skipping diseases for 1539 due to id = NaN\n",
      "skipping geographicOrigin for 1541 due to id = NaN\n",
      "skipping diseases for 1541 due to id = NaN\n",
      "skipping geographicOrigin for 1542 due to id = NaN\n",
      "skipping geographicOrigin for 1543 due to id = NaN\n",
      "skipping diseases for 1544 due to id = NaN\n",
      "skipping ethnicity for 1550 due to id = NaN\n",
      "skipping geographicOrigin for 1551 due to id = NaN\n",
      "skipping diseases for 1552 due to id = NaN\n",
      "skipping geographicOrigin for 1553 due to id = NaN\n",
      "skipping diseases for 1553 due to id = NaN\n",
      "skipping geographicOrigin for 1554 due to id = NaN\n",
      "skipping diseases for 1554 due to id = NaN\n",
      "skipping geographicOrigin for 1555 due to id = NaN\n",
      "skipping ethnicity for 1556 due to id = NaN\n",
      "skipping geographicOrigin for 1556 due to id = NaN\n",
      "skipping geographicOrigin for 1558 due to id = NaN\n",
      "skipping diseases for 1558 due to id = NaN\n",
      "skipping geographicOrigin for 1559 due to id = NaN\n",
      "skipping geographicOrigin for 1560 due to id = NaN\n",
      "skipping diseases for 1560 due to id = NaN\n",
      "skipping diseases for 1562 due to id = NaN\n",
      "skipping geographicOrigin for 1564 due to id = NaN\n",
      "skipping diseases for 1566 due to id = NaN\n",
      "skipping ethnicity for 1567 due to id = NaN\n",
      "skipping geographicOrigin for 1567 due to id = NaN\n",
      "skipping geographicOrigin for 1568 due to id = NaN\n",
      "skipping diseases for 1570 due to id = NaN\n",
      "skipping geographicOrigin for 1571 due to id = NaN\n",
      "skipping diseases for 1571 due to id = NaN\n",
      "skipping geographicOrigin for 1572 due to id = NaN\n",
      "skipping diseases for 1573 due to id = NaN\n",
      "skipping diseases for 1574 due to id = NaN\n",
      "skipping geographicOrigin for 1575 due to id = NaN\n",
      "skipping diseases for 1576 due to id = NaN\n",
      "skipping diseases for 1577 due to id = NaN\n",
      "skipping geographicOrigin for 1578 due to id = NaN\n",
      "skipping geographicOrigin for 1580 due to id = NaN\n",
      "skipping diseases for 1581 due to id = NaN\n",
      "skipping geographicOrigin for 1583 due to id = NaN\n",
      "skipping diseases for 1583 due to id = NaN\n",
      "skipping geographicOrigin for 1584 due to id = NaN\n",
      "skipping diseases for 1585 due to id = NaN\n",
      "skipping geographicOrigin for 1586 due to id = NaN\n",
      "skipping diseases for 1587 due to id = NaN\n",
      "skipping geographicOrigin for 1588 due to id = NaN\n",
      "skipping geographicOrigin for 1589 due to id = NaN\n",
      "skipping geographicOrigin for 1591 due to id = NaN\n",
      "skipping geographicOrigin for 1592 due to id = NaN\n",
      "skipping geographicOrigin for 1594 due to id = NaN\n",
      "skipping geographicOrigin for 1595 due to id = NaN\n",
      "skipping geographicOrigin for 1597 due to id = NaN\n",
      "skipping diseases for 1597 due to id = NaN\n",
      "skipping ethnicity for 1598 due to id = NaN\n",
      "skipping geographicOrigin for 1598 due to id = NaN\n",
      "skipping diseases for 1599 due to id = NaN\n",
      "skipping geographicOrigin for 1601 due to id = NaN\n",
      "skipping diseases for 1601 due to id = NaN\n",
      "skipping diseases for 1602 due to id = NaN\n",
      "skipping geographicOrigin for 1606 due to id = NaN\n",
      "skipping diseases for 1606 due to id = NaN\n",
      "skipping geographicOrigin for 1607 due to id = NaN\n",
      "skipping diseases for 1607 due to id = NaN\n",
      "skipping geographicOrigin for 1608 due to id = NaN\n",
      "skipping diseases for 1609 due to id = NaN\n",
      "skipping diseases for 1611 due to id = NaN\n",
      "skipping geographicOrigin for 1612 due to id = NaN\n",
      "skipping diseases for 1613 due to id = NaN\n",
      "skipping geographicOrigin for 1614 due to id = NaN\n",
      "skipping diseases for 1614 due to id = NaN\n",
      "skipping diseases for 1615 due to id = NaN\n",
      "skipping geographicOrigin for 1616 due to id = NaN\n",
      "skipping geographicOrigin for 1617 due to id = NaN\n",
      "skipping geographicOrigin for 1618 due to id = NaN\n",
      "skipping ethnicity for 1619 due to id = NaN\n",
      "skipping geographicOrigin for 1619 due to id = NaN\n",
      "skipping diseases for 1619 due to id = NaN\n",
      "skipping ethnicity for 1623 due to id = NaN\n",
      "skipping geographicOrigin for 1625 due to id = NaN\n",
      "skipping geographicOrigin for 1628 due to id = NaN\n",
      "skipping diseases for 1628 due to id = NaN\n",
      "skipping diseases for 1629 due to id = NaN\n",
      "skipping geographicOrigin for 1630 due to id = NaN\n",
      "skipping ethnicity for 1631 due to id = NaN\n",
      "skipping geographicOrigin for 1631 due to id = NaN\n",
      "skipping geographicOrigin for 1633 due to id = NaN\n",
      "skipping geographicOrigin for 1635 due to id = NaN\n",
      "skipping diseases for 1636 due to id = NaN\n",
      "skipping geographicOrigin for 1638 due to id = NaN\n",
      "skipping geographicOrigin for 1639 due to id = NaN\n",
      "skipping diseases for 1641 due to id = NaN\n",
      "skipping geographicOrigin for 1643 due to id = NaN\n",
      "skipping geographicOrigin for 1644 due to id = NaN\n",
      "skipping geographicOrigin for 1646 due to id = NaN\n",
      "skipping diseases for 1646 due to id = NaN\n",
      "skipping diseases for 1647 due to id = NaN\n",
      "skipping diseases for 1648 due to id = NaN\n",
      "skipping ethnicity for 1650 due to id = NaN\n",
      "skipping geographicOrigin for 1651 due to id = NaN\n",
      "skipping diseases for 1651 due to id = NaN\n",
      "skipping geographicOrigin for 1654 due to id = NaN\n",
      "skipping geographicOrigin for 1655 due to id = NaN\n",
      "skipping diseases for 1655 due to id = NaN\n",
      "skipping ethnicity for 1656 due to id = NaN\n",
      "skipping geographicOrigin for 1656 due to id = NaN\n",
      "skipping diseases for 1656 due to id = NaN\n",
      "skipping geographicOrigin for 1658 due to id = NaN\n",
      "skipping diseases for 1659 due to id = NaN\n",
      "skipping diseases for 1660 due to id = NaN\n",
      "skipping geographicOrigin for 1662 due to id = NaN\n",
      "skipping diseases for 1662 due to id = NaN\n",
      "skipping geographicOrigin for 1663 due to id = NaN\n",
      "skipping ethnicity for 1667 due to id = NaN\n",
      "skipping ethnicity for 1668 due to id = NaN\n",
      "skipping geographicOrigin for 1668 due to id = NaN\n",
      "skipping geographicOrigin for 1670 due to id = NaN\n",
      "skipping ethnicity for 1671 due to id = NaN\n",
      "skipping diseases for 1671 due to id = NaN\n",
      "skipping diseases for 1673 due to id = NaN\n",
      "skipping diseases for 1674 due to id = NaN\n",
      "skipping geographicOrigin for 1675 due to id = NaN\n",
      "skipping diseases for 1676 due to id = NaN\n",
      "skipping geographicOrigin for 1678 due to id = NaN\n",
      "skipping geographicOrigin for 1679 due to id = NaN\n",
      "skipping ethnicity for 1681 due to id = NaN\n",
      "skipping geographicOrigin for 1681 due to id = NaN\n",
      "skipping ethnicity for 1682 due to id = NaN\n",
      "skipping geographicOrigin for 1682 due to id = NaN\n",
      "skipping diseases for 1684 due to id = NaN\n",
      "skipping geographicOrigin for 1686 due to id = NaN\n",
      "skipping geographicOrigin for 1687 due to id = NaN\n",
      "skipping diseases for 1687 due to id = NaN\n",
      "skipping geographicOrigin for 1688 due to id = NaN\n",
      "skipping geographicOrigin for 1689 due to id = NaN\n",
      "skipping ethnicity for 1690 due to id = NaN\n",
      "skipping diseases for 1690 due to id = NaN\n",
      "skipping geographicOrigin for 1691 due to id = NaN\n",
      "skipping diseases for 1691 due to id = NaN\n",
      "skipping geographicOrigin for 1692 due to id = NaN\n",
      "skipping diseases for 1693 due to id = NaN\n",
      "skipping geographicOrigin for 1695 due to id = NaN\n",
      "skipping diseases for 1698 due to id = NaN\n",
      "skipping geographicOrigin for 1701 due to id = NaN\n",
      "skipping diseases for 1701 due to id = NaN\n",
      "skipping geographicOrigin for 1703 due to id = NaN\n",
      "skipping diseases for 1703 due to id = NaN\n",
      "skipping ethnicity for 1706 due to id = NaN\n",
      "skipping geographicOrigin for 1706 due to id = NaN\n",
      "skipping diseases for 1708 due to id = NaN\n",
      "skipping diseases for 1709 due to id = NaN\n",
      "skipping geographicOrigin for 1710 due to id = NaN\n",
      "skipping geographicOrigin for 1712 due to id = NaN\n",
      "skipping diseases for 1712 due to id = NaN\n",
      "skipping diseases for 1713 due to id = NaN\n",
      "skipping geographicOrigin for 1714 due to id = NaN\n",
      "skipping geographicOrigin for 1715 due to id = NaN\n",
      "skipping ethnicity for 1717 due to id = NaN\n",
      "skipping diseases for 1717 due to id = NaN\n",
      "skipping geographicOrigin for 1718 due to id = NaN\n",
      "skipping diseases for 1719 due to id = NaN\n",
      "skipping geographicOrigin for 1721 due to id = NaN\n",
      "skipping diseases for 1721 due to id = NaN\n",
      "skipping geographicOrigin for 1723 due to id = NaN\n",
      "skipping diseases for 1723 due to id = NaN\n",
      "skipping ethnicity for 1724 due to id = NaN\n",
      "skipping diseases for 1724 due to id = NaN\n",
      "skipping diseases for 1725 due to id = NaN\n",
      "skipping geographicOrigin for 1727 due to id = NaN\n",
      "skipping diseases for 1727 due to id = NaN\n",
      "skipping diseases for 1728 due to id = NaN\n",
      "skipping diseases for 1729 due to id = NaN\n",
      "skipping geographicOrigin for 1732 due to id = NaN\n",
      "skipping diseases for 1733 due to id = NaN\n",
      "skipping geographicOrigin for 1734 due to id = NaN\n",
      "skipping diseases for 1735 due to id = NaN\n",
      "skipping geographicOrigin for 1737 due to id = NaN\n",
      "skipping diseases for 1738 due to id = NaN\n",
      "skipping ethnicity for 1739 due to id = NaN\n",
      "skipping diseases for 1739 due to id = NaN\n",
      "skipping geographicOrigin for 1740 due to id = NaN\n",
      "skipping ethnicity for 1741 due to id = NaN\n",
      "skipping geographicOrigin for 1741 due to id = NaN\n",
      "skipping diseases for 1741 due to id = NaN\n",
      "skipping geographicOrigin for 1742 due to id = NaN\n",
      "skipping diseases for 1744 due to id = NaN\n",
      "skipping geographicOrigin for 1746 due to id = NaN\n",
      "skipping diseases for 1746 due to id = NaN\n",
      "skipping diseases for 1747 due to id = NaN\n",
      "skipping diseases for 1748 due to id = NaN\n",
      "skipping diseases for 1750 due to id = NaN\n",
      "skipping diseases for 1751 due to id = NaN\n",
      "skipping ethnicity for 1753 due to id = NaN\n",
      "skipping geographicOrigin for 1753 due to id = NaN\n",
      "skipping geographicOrigin for 1755 due to id = NaN\n",
      "skipping diseases for 1755 due to id = NaN\n",
      "skipping diseases for 1756 due to id = NaN\n",
      "skipping diseases for 1760 due to id = NaN\n",
      "skipping geographicOrigin for 1762 due to id = NaN\n",
      "skipping diseases for 1762 due to id = NaN\n",
      "skipping geographicOrigin for 1767 due to id = NaN\n",
      "skipping diseases for 1767 due to id = NaN\n",
      "skipping geographicOrigin for 1770 due to id = NaN\n",
      "skipping ethnicity for 1771 due to id = NaN\n",
      "skipping geographicOrigin for 1771 due to id = NaN\n",
      "skipping geographicOrigin for 1772 due to id = NaN\n",
      "skipping geographicOrigin for 1773 due to id = NaN\n",
      "skipping ethnicity for 1774 due to id = NaN\n",
      "skipping geographicOrigin for 1774 due to id = NaN\n",
      "skipping geographicOrigin for 1776 due to id = NaN\n",
      "skipping geographicOrigin for 1777 due to id = NaN\n",
      "skipping diseases for 1777 due to id = NaN\n",
      "skipping ethnicity for 1778 due to id = NaN\n",
      "skipping geographicOrigin for 1778 due to id = NaN\n",
      "skipping ethnicity for 1780 due to id = NaN\n",
      "skipping geographicOrigin for 1780 due to id = NaN\n",
      "skipping diseases for 1780 due to id = NaN\n",
      "skipping geographicOrigin for 1781 due to id = NaN\n",
      "skipping diseases for 1781 due to id = NaN\n",
      "skipping diseases for 1782 due to id = NaN\n",
      "skipping geographicOrigin for 1783 due to id = NaN\n",
      "skipping ethnicity for 1784 due to id = NaN\n",
      "skipping geographicOrigin for 1784 due to id = NaN\n",
      "skipping diseases for 1784 due to id = NaN\n",
      "skipping diseases for 1786 due to id = NaN\n",
      "skipping geographicOrigin for 1789 due to id = NaN\n",
      "skipping geographicOrigin for 1790 due to id = NaN\n",
      "skipping geographicOrigin for 1794 due to id = NaN\n",
      "skipping diseases for 1797 due to id = NaN\n",
      "skipping diseases for 1800 due to id = NaN\n",
      "skipping diseases for 1801 due to id = NaN\n",
      "skipping geographicOrigin for 1802 due to id = NaN\n",
      "skipping geographicOrigin for 1807 due to id = NaN\n",
      "skipping geographicOrigin for 1810 due to id = NaN\n",
      "skipping diseases for 1810 due to id = NaN\n",
      "skipping ethnicity for 1812 due to id = NaN\n",
      "skipping geographicOrigin for 1812 due to id = NaN\n",
      "skipping geographicOrigin for 1813 due to id = NaN\n",
      "skipping geographicOrigin for 1815 due to id = NaN\n",
      "skipping diseases for 1817 due to id = NaN\n",
      "skipping diseases for 1819 due to id = NaN\n",
      "skipping geographicOrigin for 1823 due to id = NaN\n",
      "skipping diseases for 1823 due to id = NaN\n",
      "skipping geographicOrigin for 1825 due to id = NaN\n",
      "skipping geographicOrigin for 1826 due to id = NaN\n",
      "skipping diseases for 1826 due to id = NaN\n",
      "skipping geographicOrigin for 1829 due to id = NaN\n",
      "skipping ethnicity for 1830 due to id = NaN\n",
      "skipping diseases for 1831 due to id = NaN\n",
      "skipping geographicOrigin for 1832 due to id = NaN\n",
      "skipping diseases for 1832 due to id = NaN\n",
      "skipping ethnicity for 1833 due to id = NaN\n",
      "skipping geographicOrigin for 1834 due to id = NaN\n",
      "skipping ethnicity for 1835 due to id = NaN\n",
      "skipping ethnicity for 1837 due to id = NaN\n",
      "skipping geographicOrigin for 1837 due to id = NaN\n",
      "skipping ethnicity for 1838 due to id = NaN\n",
      "skipping geographicOrigin for 1838 due to id = NaN\n",
      "skipping geographicOrigin for 1841 due to id = NaN\n",
      "skipping geographicOrigin for 1842 due to id = NaN\n",
      "skipping geographicOrigin for 1843 due to id = NaN\n",
      "skipping diseases for 1845 due to id = NaN\n",
      "skipping geographicOrigin for 1846 due to id = NaN\n",
      "skipping geographicOrigin for 1847 due to id = NaN\n",
      "skipping geographicOrigin for 1848 due to id = NaN\n",
      "skipping geographicOrigin for 1849 due to id = NaN\n",
      "skipping diseases for 1851 due to id = NaN\n",
      "skipping ethnicity for 1852 due to id = NaN\n",
      "skipping geographicOrigin for 1852 due to id = NaN\n",
      "skipping diseases for 1852 due to id = NaN\n",
      "skipping geographicOrigin for 1854 due to id = NaN\n",
      "skipping diseases for 1856 due to id = NaN\n",
      "skipping diseases for 1857 due to id = NaN\n",
      "skipping ethnicity for 1858 due to id = NaN\n",
      "skipping geographicOrigin for 1858 due to id = NaN\n",
      "skipping geographicOrigin for 1859 due to id = NaN\n",
      "skipping diseases for 1859 due to id = NaN\n",
      "skipping geographicOrigin for 1861 due to id = NaN\n",
      "skipping geographicOrigin for 1862 due to id = NaN\n",
      "skipping diseases for 1862 due to id = NaN\n",
      "skipping geographicOrigin for 1863 due to id = NaN\n",
      "skipping diseases for 1865 due to id = NaN\n",
      "skipping geographicOrigin for 1866 due to id = NaN\n",
      "skipping diseases for 1866 due to id = NaN\n",
      "skipping diseases for 1867 due to id = NaN\n",
      "skipping geographicOrigin for 1868 due to id = NaN\n",
      "skipping geographicOrigin for 1869 due to id = NaN\n",
      "skipping ethnicity for 1870 due to id = NaN\n",
      "skipping geographicOrigin for 1871 due to id = NaN\n",
      "skipping diseases for 1873 due to id = NaN\n",
      "skipping diseases for 1874 due to id = NaN\n",
      "skipping diseases for 1877 due to id = NaN\n",
      "skipping geographicOrigin for 1878 due to id = NaN\n",
      "skipping diseases for 1880 due to id = NaN\n",
      "skipping geographicOrigin for 1881 due to id = NaN\n",
      "skipping geographicOrigin for 1883 due to id = NaN\n",
      "skipping diseases for 1885 due to id = NaN\n",
      "skipping diseases for 1887 due to id = NaN\n",
      "skipping diseases for 1889 due to id = NaN\n",
      "skipping diseases for 1890 due to id = NaN\n",
      "skipping geographicOrigin for 1891 due to id = NaN\n",
      "skipping diseases for 1893 due to id = NaN\n",
      "skipping geographicOrigin for 1894 due to id = NaN\n",
      "skipping diseases for 1894 due to id = NaN\n",
      "skipping geographicOrigin for 1896 due to id = NaN\n",
      "skipping diseases for 1897 due to id = NaN\n",
      "skipping geographicOrigin for 1898 due to id = NaN\n",
      "skipping geographicOrigin for 1899 due to id = NaN\n",
      "skipping diseases for 1900 due to id = NaN\n",
      "skipping ethnicity for 1903 due to id = NaN\n",
      "skipping diseases for 1903 due to id = NaN\n",
      "skipping geographicOrigin for 1904 due to id = NaN\n",
      "skipping geographicOrigin for 1906 due to id = NaN\n",
      "skipping diseases for 1906 due to id = NaN\n",
      "skipping diseases for 1909 due to id = NaN\n",
      "skipping geographicOrigin for 1910 due to id = NaN\n",
      "skipping geographicOrigin for 1911 due to id = NaN\n",
      "skipping geographicOrigin for 1914 due to id = NaN\n",
      "skipping diseases for 1914 due to id = NaN\n",
      "skipping geographicOrigin for 1915 due to id = NaN\n",
      "skipping geographicOrigin for 1916 due to id = NaN\n",
      "skipping diseases for 1917 due to id = NaN\n",
      "skipping geographicOrigin for 1919 due to id = NaN\n",
      "skipping geographicOrigin for 1922 due to id = NaN\n",
      "skipping ethnicity for 1924 due to id = NaN\n",
      "skipping geographicOrigin for 1924 due to id = NaN\n",
      "skipping geographicOrigin for 1926 due to id = NaN\n",
      "skipping diseases for 1928 due to id = NaN\n",
      "skipping diseases for 1929 due to id = NaN\n",
      "skipping ethnicity for 1931 due to id = NaN\n",
      "skipping geographicOrigin for 1932 due to id = NaN\n",
      "skipping geographicOrigin for 1934 due to id = NaN\n",
      "skipping geographicOrigin for 1935 due to id = NaN\n",
      "skipping geographicOrigin for 1936 due to id = NaN\n",
      "skipping diseases for 1936 due to id = NaN\n",
      "skipping ethnicity for 1942 due to id = NaN\n",
      "skipping geographicOrigin for 1942 due to id = NaN\n",
      "skipping geographicOrigin for 1943 due to id = NaN\n",
      "skipping geographicOrigin for 1944 due to id = NaN\n",
      "skipping diseases for 1944 due to id = NaN\n",
      "skipping diseases for 1945 due to id = NaN\n",
      "skipping geographicOrigin for 1946 due to id = NaN\n",
      "skipping diseases for 1946 due to id = NaN\n",
      "skipping diseases for 1947 due to id = NaN\n",
      "skipping ethnicity for 1948 due to id = NaN\n",
      "skipping geographicOrigin for 1949 due to id = NaN\n",
      "skipping geographicOrigin for 1951 due to id = NaN\n",
      "skipping diseases for 1951 due to id = NaN\n",
      "skipping diseases for 1954 due to id = NaN\n",
      "skipping diseases for 1955 due to id = NaN\n",
      "skipping geographicOrigin for 1958 due to id = NaN\n",
      "skipping geographicOrigin for 1961 due to id = NaN\n",
      "skipping diseases for 1961 due to id = NaN\n",
      "skipping geographicOrigin for 1963 due to id = NaN\n",
      "skipping geographicOrigin for 1966 due to id = NaN\n",
      "skipping geographicOrigin for 1968 due to id = NaN\n",
      "skipping geographicOrigin for 1970 due to id = NaN\n",
      "skipping geographicOrigin for 1972 due to id = NaN\n",
      "skipping diseases for 1972 due to id = NaN\n",
      "skipping geographicOrigin for 1973 due to id = NaN\n",
      "skipping diseases for 1973 due to id = NaN\n",
      "skipping geographicOrigin for 1974 due to id = NaN\n",
      "skipping diseases for 1976 due to id = NaN\n",
      "skipping geographicOrigin for 1977 due to id = NaN\n",
      "skipping diseases for 1977 due to id = NaN\n",
      "skipping geographicOrigin for 1978 due to id = NaN\n",
      "skipping diseases for 1978 due to id = NaN\n",
      "skipping diseases for 1983 due to id = NaN\n",
      "skipping geographicOrigin for 1986 due to id = NaN\n",
      "skipping geographicOrigin for 1987 due to id = NaN\n",
      "skipping diseases for 1987 due to id = NaN\n",
      "skipping geographicOrigin for 1992 due to id = NaN\n",
      "skipping diseases for 1992 due to id = NaN\n",
      "skipping diseases for 1994 due to id = NaN\n",
      "skipping diseases for 1995 due to id = NaN\n",
      "skipping ethnicity for 2000 due to id = NaN\n",
      "skipping geographicOrigin for 2003 due to id = NaN\n",
      "skipping geographicOrigin for 2007 due to id = NaN\n",
      "skipping diseases for 2007 due to id = NaN\n",
      "skipping geographicOrigin for 2011 due to id = NaN\n",
      "skipping ethnicity for 2012 due to id = NaN\n",
      "skipping geographicOrigin for 2012 due to id = NaN\n",
      "skipping diseases for 2012 due to id = NaN\n",
      "skipping ethnicity for 2014 due to id = NaN\n",
      "skipping diseases for 2014 due to id = NaN\n",
      "skipping geographicOrigin for 2016 due to id = NaN\n",
      "skipping geographicOrigin for 2018 due to id = NaN\n",
      "skipping geographicOrigin for 2022 due to id = NaN\n",
      "skipping diseases for 2022 due to id = NaN\n",
      "skipping geographicOrigin for 2024 due to id = NaN\n",
      "skipping diseases for 2024 due to id = NaN\n",
      "skipping geographicOrigin for 2026 due to id = NaN\n",
      "skipping ethnicity for 2029 due to id = NaN\n",
      "skipping geographicOrigin for 2033 due to id = NaN\n",
      "skipping geographicOrigin for 2038 due to id = NaN\n",
      "skipping diseases for 2038 due to id = NaN\n",
      "skipping geographicOrigin for 2039 due to id = NaN\n",
      "skipping diseases for 2039 due to id = NaN\n",
      "skipping geographicOrigin for 2040 due to id = NaN\n",
      "skipping diseases for 2041 due to id = NaN\n",
      "skipping geographicOrigin for 2043 due to id = NaN\n",
      "skipping geographicOrigin for 2044 due to id = NaN\n",
      "skipping ethnicity for 2045 due to id = NaN\n",
      "skipping geographicOrigin for 2046 due to id = NaN\n",
      "skipping geographicOrigin for 2048 due to id = NaN\n",
      "skipping geographicOrigin for 2049 due to id = NaN\n",
      "skipping diseases for 2049 due to id = NaN\n",
      "skipping geographicOrigin for 2053 due to id = NaN\n",
      "skipping geographicOrigin for 2054 due to id = NaN\n",
      "skipping geographicOrigin for 2055 due to id = NaN\n",
      "skipping diseases for 2055 due to id = NaN\n",
      "skipping geographicOrigin for 2056 due to id = NaN\n",
      "skipping diseases for 2056 due to id = NaN\n",
      "skipping geographicOrigin for 2057 due to id = NaN\n",
      "skipping geographicOrigin for 2059 due to id = NaN\n",
      "skipping geographicOrigin for 2060 due to id = NaN\n",
      "skipping geographicOrigin for 2062 due to id = NaN\n",
      "skipping diseases for 2062 due to id = NaN\n",
      "skipping geographicOrigin for 2063 due to id = NaN\n",
      "skipping geographicOrigin for 2064 due to id = NaN\n",
      "skipping diseases for 2064 due to id = NaN\n",
      "skipping geographicOrigin for 2065 due to id = NaN\n",
      "skipping geographicOrigin for 2066 due to id = NaN\n",
      "skipping ethnicity for 2067 due to id = NaN\n",
      "skipping geographicOrigin for 2067 due to id = NaN\n",
      "skipping diseases for 2067 due to id = NaN\n",
      "skipping diseases for 2068 due to id = NaN\n",
      "skipping geographicOrigin for 2069 due to id = NaN\n",
      "skipping diseases for 2070 due to id = NaN\n",
      "skipping ethnicity for 2071 due to id = NaN\n",
      "skipping geographicOrigin for 2071 due to id = NaN\n",
      "skipping diseases for 2071 due to id = NaN\n",
      "skipping geographicOrigin for 2073 due to id = NaN\n",
      "skipping diseases for 2075 due to id = NaN\n",
      "skipping ethnicity for 2076 due to id = NaN\n",
      "skipping geographicOrigin for 2076 due to id = NaN\n",
      "skipping geographicOrigin for 2078 due to id = NaN\n",
      "skipping geographicOrigin for 2080 due to id = NaN\n",
      "skipping geographicOrigin for 2084 due to id = NaN\n",
      "skipping diseases for 2084 due to id = NaN\n",
      "skipping diseases for 2086 due to id = NaN\n",
      "skipping ethnicity for 2087 due to id = NaN\n",
      "skipping diseases for 2087 due to id = NaN\n",
      "skipping diseases for 2088 due to id = NaN\n",
      "skipping geographicOrigin for 2092 due to id = NaN\n",
      "skipping ethnicity for 2093 due to id = NaN\n",
      "skipping diseases for 2095 due to id = NaN\n",
      "skipping ethnicity for 2097 due to id = NaN\n",
      "skipping geographicOrigin for 2098 due to id = NaN\n",
      "skipping geographicOrigin for 2101 due to id = NaN\n",
      "skipping diseases for 2101 due to id = NaN\n",
      "skipping geographicOrigin for 2102 due to id = NaN\n",
      "skipping geographicOrigin for 2103 due to id = NaN\n",
      "skipping diseases for 2103 due to id = NaN\n",
      "skipping diseases for 2104 due to id = NaN\n",
      "skipping diseases for 2105 due to id = NaN\n",
      "skipping geographicOrigin for 2107 due to id = NaN\n",
      "skipping diseases for 2109 due to id = NaN\n",
      "skipping diseases for 2112 due to id = NaN\n",
      "skipping diseases for 2114 due to id = NaN\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "skipping ethnicity for 2117 due to id = NaN\n",
      "skipping ethnicity for 2119 due to id = NaN\n",
      "skipping geographicOrigin for 2120 due to id = NaN\n",
      "skipping diseases for 2120 due to id = NaN\n",
      "skipping geographicOrigin for 2121 due to id = NaN\n",
      "skipping geographicOrigin for 2122 due to id = NaN\n",
      "skipping diseases for 2123 due to id = NaN\n",
      "skipping diseases for 2125 due to id = NaN\n",
      "skipping geographicOrigin for 2128 due to id = NaN\n",
      "skipping geographicOrigin for 2129 due to id = NaN\n",
      "skipping geographicOrigin for 2130 due to id = NaN\n",
      "skipping geographicOrigin for 2131 due to id = NaN\n",
      "skipping diseases for 2131 due to id = NaN\n",
      "skipping ethnicity for 2133 due to id = NaN\n",
      "skipping geographicOrigin for 2135 due to id = NaN\n",
      "skipping diseases for 2135 due to id = NaN\n",
      "skipping diseases for 2136 due to id = NaN\n",
      "skipping ethnicity for 2137 due to id = NaN\n",
      "skipping geographicOrigin for 2137 due to id = NaN\n",
      "skipping diseases for 2137 due to id = NaN\n",
      "skipping geographicOrigin for 2138 due to id = NaN\n",
      "skipping diseases for 2138 due to id = NaN\n",
      "skipping geographicOrigin for 2139 due to id = NaN\n",
      "skipping ethnicity for 2140 due to id = NaN\n",
      "skipping ethnicity for 2141 due to id = NaN\n",
      "skipping geographicOrigin for 2144 due to id = NaN\n",
      "skipping diseases for 2144 due to id = NaN\n",
      "skipping geographicOrigin for 2145 due to id = NaN\n",
      "skipping geographicOrigin for 2147 due to id = NaN\n",
      "skipping diseases for 2151 due to id = NaN\n",
      "skipping geographicOrigin for 2152 due to id = NaN\n",
      "skipping diseases for 2153 due to id = NaN\n",
      "skipping geographicOrigin for 2155 due to id = NaN\n",
      "skipping diseases for 2155 due to id = NaN\n",
      "skipping ethnicity for 2156 due to id = NaN\n",
      "skipping geographicOrigin for 2157 due to id = NaN\n",
      "skipping ethnicity for 2159 due to id = NaN\n",
      "skipping geographicOrigin for 2159 due to id = NaN\n",
      "skipping diseases for 2159 due to id = NaN\n",
      "skipping ethnicity for 2161 due to id = NaN\n",
      "skipping geographicOrigin for 2162 due to id = NaN\n",
      "skipping geographicOrigin for 2163 due to id = NaN\n",
      "skipping diseases for 2163 due to id = NaN\n",
      "skipping geographicOrigin for 2165 due to id = NaN\n",
      "skipping diseases for 2165 due to id = NaN\n",
      "skipping diseases for 2167 due to id = NaN\n",
      "skipping geographicOrigin for 2169 due to id = NaN\n",
      "skipping ethnicity for 2170 due to id = NaN\n",
      "skipping diseases for 2170 due to id = NaN\n",
      "skipping geographicOrigin for 2171 due to id = NaN\n",
      "skipping diseases for 2174 due to id = NaN\n",
      "skipping diseases for 2177 due to id = NaN\n",
      "skipping geographicOrigin for 2178 due to id = NaN\n",
      "skipping geographicOrigin for 2179 due to id = NaN\n",
      "skipping geographicOrigin for 2182 due to id = NaN\n",
      "skipping geographicOrigin for 2185 due to id = NaN\n",
      "skipping geographicOrigin for 2186 due to id = NaN\n",
      "skipping geographicOrigin for 2187 due to id = NaN\n",
      "skipping diseases for 2190 due to id = NaN\n",
      "skipping geographicOrigin for 2191 due to id = NaN\n",
      "skipping ethnicity for 2193 due to id = NaN\n",
      "skipping ethnicity for 2194 due to id = NaN\n",
      "skipping ethnicity for 2196 due to id = NaN\n",
      "skipping diseases for 2200 due to id = NaN\n",
      "skipping geographicOrigin for 2202 due to id = NaN\n",
      "skipping geographicOrigin for 2204 due to id = NaN\n",
      "skipping diseases for 2204 due to id = NaN\n",
      "skipping geographicOrigin for 2205 due to id = NaN\n",
      "skipping geographicOrigin for 2206 due to id = NaN\n",
      "skipping geographicOrigin for 2207 due to id = NaN\n",
      "skipping geographicOrigin for 2208 due to id = NaN\n",
      "skipping diseases for 2208 due to id = NaN\n",
      "skipping geographicOrigin for 2209 due to id = NaN\n",
      "skipping diseases for 2209 due to id = NaN\n",
      "skipping geographicOrigin for 2212 due to id = NaN\n",
      "skipping diseases for 2212 due to id = NaN\n",
      "skipping geographicOrigin for 2213 due to id = NaN\n",
      "skipping geographicOrigin for 2214 due to id = NaN\n",
      "skipping diseases for 2214 due to id = NaN\n",
      "skipping geographicOrigin for 2215 due to id = NaN\n",
      "skipping diseases for 2215 due to id = NaN\n",
      "skipping ethnicity for 2216 due to id = NaN\n",
      "skipping diseases for 2218 due to id = NaN\n",
      "skipping diseases for 2219 due to id = NaN\n",
      "skipping diseases for 2221 due to id = NaN\n",
      "skipping geographicOrigin for 2224 due to id = NaN\n",
      "skipping diseases for 2225 due to id = NaN\n",
      "skipping geographicOrigin for 2228 due to id = NaN\n",
      "skipping diseases for 2228 due to id = NaN\n",
      "skipping geographicOrigin for 2231 due to id = NaN\n",
      "skipping geographicOrigin for 2232 due to id = NaN\n",
      "skipping diseases for 2233 due to id = NaN\n",
      "skipping diseases for 2234 due to id = NaN\n",
      "skipping geographicOrigin for 2235 due to id = NaN\n",
      "skipping diseases for 2237 due to id = NaN\n",
      "skipping geographicOrigin for 2239 due to id = NaN\n",
      "skipping geographicOrigin for 2240 due to id = NaN\n",
      "skipping geographicOrigin for 2241 due to id = NaN\n",
      "skipping diseases for 2241 due to id = NaN\n",
      "skipping geographicOrigin for 2242 due to id = NaN\n",
      "skipping diseases for 2242 due to id = NaN\n",
      "skipping geographicOrigin for 2243 due to id = NaN\n",
      "skipping diseases for 2244 due to id = NaN\n",
      "skipping geographicOrigin for 2247 due to id = NaN\n",
      "skipping geographicOrigin for 2249 due to id = NaN\n",
      "skipping diseases for 2249 due to id = NaN\n",
      "skipping ethnicity for 2251 due to id = NaN\n",
      "skipping diseases for 2251 due to id = NaN\n",
      "skipping diseases for 2252 due to id = NaN\n",
      "skipping geographicOrigin for 2255 due to id = NaN\n",
      "skipping diseases for 2256 due to id = NaN\n",
      "skipping geographicOrigin for 2257 due to id = NaN\n",
      "skipping ethnicity for 2258 due to id = NaN\n",
      "skipping ethnicity for 2259 due to id = NaN\n",
      "skipping diseases for 2259 due to id = NaN\n",
      "skipping diseases for 2260 due to id = NaN\n",
      "skipping ethnicity for 2262 due to id = NaN\n",
      "skipping geographicOrigin for 2262 due to id = NaN\n",
      "skipping geographicOrigin for 2264 due to id = NaN\n",
      "skipping diseases for 2264 due to id = NaN\n",
      "skipping diseases for 2270 due to id = NaN\n",
      "skipping diseases for 2272 due to id = NaN\n",
      "skipping ethnicity for 2273 due to id = NaN\n",
      "skipping geographicOrigin for 2273 due to id = NaN\n",
      "skipping geographicOrigin for 2274 due to id = NaN\n",
      "skipping geographicOrigin for 2276 due to id = NaN\n",
      "skipping geographicOrigin for 2280 due to id = NaN\n",
      "skipping geographicOrigin for 2283 due to id = NaN\n",
      "skipping diseases for 2283 due to id = NaN\n",
      "skipping geographicOrigin for 2285 due to id = NaN\n",
      "skipping diseases for 2286 due to id = NaN\n",
      "skipping diseases for 2287 due to id = NaN\n",
      "skipping geographicOrigin for 2289 due to id = NaN\n",
      "skipping diseases for 2289 due to id = NaN\n",
      "skipping diseases for 2292 due to id = NaN\n",
      "skipping ethnicity for 2296 due to id = NaN\n",
      "skipping geographicOrigin for 2296 due to id = NaN\n",
      "skipping diseases for 2297 due to id = NaN\n",
      "skipping geographicOrigin for 2298 due to id = NaN\n",
      "skipping diseases for 2298 due to id = NaN\n",
      "skipping diseases for 2302 due to id = NaN\n",
      "skipping geographicOrigin for 2303 due to id = NaN\n",
      "skipping geographicOrigin for 2304 due to id = NaN\n",
      "skipping geographicOrigin for 2307 due to id = NaN\n",
      "skipping diseases for 2307 due to id = NaN\n",
      "skipping geographicOrigin for 2308 due to id = NaN\n",
      "skipping geographicOrigin for 2310 due to id = NaN\n",
      "skipping diseases for 2310 due to id = NaN\n",
      "skipping geographicOrigin for 2311 due to id = NaN\n",
      "skipping diseases for 2311 due to id = NaN\n",
      "skipping diseases for 2314 due to id = NaN\n",
      "skipping diseases for 2318 due to id = NaN\n",
      "skipping diseases for 2319 due to id = NaN\n",
      "skipping geographicOrigin for 2320 due to id = NaN\n",
      "skipping geographicOrigin for 2321 due to id = NaN\n",
      "skipping diseases for 2323 due to id = NaN\n",
      "skipping ethnicity for 2325 due to id = NaN\n",
      "skipping geographicOrigin for 2325 due to id = NaN\n",
      "skipping diseases for 2325 due to id = NaN\n",
      "skipping geographicOrigin for 2327 due to id = NaN\n",
      "skipping diseases for 2328 due to id = NaN\n",
      "skipping geographicOrigin for 2329 due to id = NaN\n",
      "skipping diseases for 2330 due to id = NaN\n",
      "skipping diseases for 2331 due to id = NaN\n",
      "skipping geographicOrigin for 2333 due to id = NaN\n",
      "skipping geographicOrigin for 2335 due to id = NaN\n",
      "skipping diseases for 2336 due to id = NaN\n",
      "skipping diseases for 2337 due to id = NaN\n",
      "skipping ethnicity for 2339 due to id = NaN\n",
      "skipping geographicOrigin for 2339 due to id = NaN\n",
      "skipping ethnicity for 2342 due to id = NaN\n",
      "skipping geographicOrigin for 2342 due to id = NaN\n",
      "skipping diseases for 2342 due to id = NaN\n",
      "skipping geographicOrigin for 2344 due to id = NaN\n",
      "skipping geographicOrigin for 2345 due to id = NaN\n",
      "skipping geographicOrigin for 2348 due to id = NaN\n",
      "skipping geographicOrigin for 2349 due to id = NaN\n",
      "skipping geographicOrigin for 2350 due to id = NaN\n",
      "skipping geographicOrigin for 2351 due to id = NaN\n",
      "skipping geographicOrigin for 2357 due to id = NaN\n",
      "skipping diseases for 2357 due to id = NaN\n",
      "skipping ethnicity for 2358 due to id = NaN\n",
      "skipping geographicOrigin for 2358 due to id = NaN\n",
      "skipping diseases for 2358 due to id = NaN\n",
      "skipping geographicOrigin for 2359 due to id = NaN\n",
      "skipping geographicOrigin for 2361 due to id = NaN\n",
      "skipping diseases for 2363 due to id = NaN\n",
      "skipping geographicOrigin for 2364 due to id = NaN\n",
      "skipping ethnicity for 2365 due to id = NaN\n",
      "skipping geographicOrigin for 2366 due to id = NaN\n",
      "skipping diseases for 2367 due to id = NaN\n",
      "skipping geographicOrigin for 2369 due to id = NaN\n",
      "skipping diseases for 2369 due to id = NaN\n",
      "skipping diseases for 2370 due to id = NaN\n",
      "skipping geographicOrigin for 2371 due to id = NaN\n",
      "skipping diseases for 2371 due to id = NaN\n",
      "skipping diseases for 2372 due to id = NaN\n",
      "skipping geographicOrigin for 2373 due to id = NaN\n",
      "skipping diseases for 2373 due to id = NaN\n",
      "skipping geographicOrigin for 2375 due to id = NaN\n",
      "skipping geographicOrigin for 2376 due to id = NaN\n",
      "skipping diseases for 2376 due to id = NaN\n",
      "skipping ethnicity for 2377 due to id = NaN\n",
      "skipping geographicOrigin for 2380 due to id = NaN\n",
      "skipping geographicOrigin for 2382 due to id = NaN\n",
      "skipping diseases for 2383 due to id = NaN\n",
      "skipping ethnicity for 2385 due to id = NaN\n",
      "skipping geographicOrigin for 2386 due to id = NaN\n",
      "skipping geographicOrigin for 2387 due to id = NaN\n",
      "skipping diseases for 2387 due to id = NaN\n",
      "skipping diseases for 2390 due to id = NaN\n",
      "skipping geographicOrigin for 2391 due to id = NaN\n",
      "skipping geographicOrigin for 2392 due to id = NaN\n",
      "skipping diseases for 2393 due to id = NaN\n",
      "skipping geographicOrigin for 2394 due to id = NaN\n",
      "skipping geographicOrigin for 2395 due to id = NaN\n",
      "skipping geographicOrigin for 2396 due to id = NaN\n",
      "skipping diseases for 2396 due to id = NaN\n",
      "skipping geographicOrigin for 2398 due to id = NaN\n",
      "skipping diseases for 2400 due to id = NaN\n",
      "skipping diseases for 2401 due to id = NaN\n",
      "skipping geographicOrigin for 2404 due to id = NaN\n",
      "skipping diseases for 2404 due to id = NaN\n",
      "skipping ethnicity for 2410 due to id = NaN\n",
      "skipping geographicOrigin for 2411 due to id = NaN\n",
      "skipping diseases for 2411 due to id = NaN\n",
      "skipping geographicOrigin for 2412 due to id = NaN\n",
      "skipping ethnicity for 2413 due to id = NaN\n",
      "skipping geographicOrigin for 2413 due to id = NaN\n",
      "skipping geographicOrigin for 2414 due to id = NaN\n",
      "skipping geographicOrigin for 2415 due to id = NaN\n",
      "skipping ethnicity for 2417 due to id = NaN\n",
      "skipping geographicOrigin for 2417 due to id = NaN\n",
      "skipping diseases for 2417 due to id = NaN\n",
      "skipping geographicOrigin for 2418 due to id = NaN\n",
      "skipping geographicOrigin for 2420 due to id = NaN\n",
      "skipping geographicOrigin for 2421 due to id = NaN\n",
      "skipping geographicOrigin for 2423 due to id = NaN\n",
      "skipping diseases for 2423 due to id = NaN\n",
      "skipping ethnicity for 2424 due to id = NaN\n",
      "skipping geographicOrigin for 2425 due to id = NaN\n",
      "skipping diseases for 2425 due to id = NaN\n",
      "skipping geographicOrigin for 2426 due to id = NaN\n",
      "skipping geographicOrigin for 2428 due to id = NaN\n",
      "skipping ethnicity for 2430 due to id = NaN\n",
      "skipping geographicOrigin for 2430 due to id = NaN\n",
      "skipping geographicOrigin for 2431 due to id = NaN\n",
      "skipping diseases for 2431 due to id = NaN\n",
      "skipping ethnicity for 2432 due to id = NaN\n",
      "skipping geographicOrigin for 2434 due to id = NaN\n",
      "skipping geographicOrigin for 2436 due to id = NaN\n",
      "skipping diseases for 2436 due to id = NaN\n",
      "skipping geographicOrigin for 2438 due to id = NaN\n",
      "skipping ethnicity for 2440 due to id = NaN\n",
      "skipping geographicOrigin for 2441 due to id = NaN\n",
      "skipping geographicOrigin for 2444 due to id = NaN\n",
      "skipping geographicOrigin for 2445 due to id = NaN\n",
      "skipping diseases for 2445 due to id = NaN\n",
      "skipping geographicOrigin for 2446 due to id = NaN\n",
      "skipping diseases for 2446 due to id = NaN\n",
      "skipping geographicOrigin for 2449 due to id = NaN\n",
      "skipping geographicOrigin for 2450 due to id = NaN\n",
      "skipping diseases for 2451 due to id = NaN\n",
      "skipping diseases for 2452 due to id = NaN\n",
      "skipping diseases for 2453 due to id = NaN\n",
      "skipping ethnicity for 2457 due to id = NaN\n",
      "skipping diseases for 2457 due to id = NaN\n",
      "skipping diseases for 2458 due to id = NaN\n",
      "skipping diseases for 2460 due to id = NaN\n",
      "skipping geographicOrigin for 2461 due to id = NaN\n",
      "skipping geographicOrigin for 2463 due to id = NaN\n",
      "skipping diseases for 2463 due to id = NaN\n",
      "skipping diseases for 2464 due to id = NaN\n",
      "skipping geographicOrigin for 2467 due to id = NaN\n",
      "skipping diseases for 2471 due to id = NaN\n",
      "skipping geographicOrigin for 2475 due to id = NaN\n",
      "skipping diseases for 2475 due to id = NaN\n",
      "skipping ethnicity for 2476 due to id = NaN\n",
      "skipping geographicOrigin for 2476 due to id = NaN\n",
      "skipping geographicOrigin for 2478 due to id = NaN\n",
      "skipping diseases for 2478 due to id = NaN\n",
      "skipping geographicOrigin for 2481 due to id = NaN\n",
      "skipping geographicOrigin for 2483 due to id = NaN\n",
      "skipping diseases for 2483 due to id = NaN\n",
      "skipping diseases for 2487 due to id = NaN\n",
      "skipping geographicOrigin for 2488 due to id = NaN\n",
      "skipping ethnicity for 2491 due to id = NaN\n",
      "skipping geographicOrigin for 2492 due to id = NaN\n",
      "skipping diseases for 2492 due to id = NaN\n",
      "skipping ethnicity for 2493 due to id = NaN\n",
      "skipping geographicOrigin for 2493 due to id = NaN\n",
      "skipping geographicOrigin for 2494 due to id = NaN\n",
      "skipping geographicOrigin for 2495 due to id = NaN\n",
      "skipping diseases for 2495 due to id = NaN\n",
      "skipping geographicOrigin for 2496 due to id = NaN\n",
      "skipping diseases for 2497 due to id = NaN\n",
      "skipping geographicOrigin for 2500 due to id = NaN\n",
      "skipping geographicOrigin for 2501 due to id = NaN\n",
      "skipping diseases for 2501 due to id = NaN\n",
      "skipping ethnicity for 2502 due to id = NaN\n",
      "skipping geographicOrigin for 2503 due to id = NaN\n"
     ]
    }
   ],
   "source": [
    "# add individuals\n",
    "for idx, row in df.iterrows():\n",
    "#for idx, row in df.head(1).iterrows():\n",
    "    # print(row)\n",
    "    individual = {}\n",
    "    individual.update(id=row['id'])\n",
    "    \n",
    "    individual.update(sex={\n",
    "        'id': row['sex.id'],\n",
    "        'label': row['sex.label']\n",
    "    })\n",
    "\n",
    "    if not isinstance(row['ethnicity.id'], float):\n",
    "        individual.update(ethnicity={\n",
    "            'id': row['ethnicity.id'],\n",
    "            'label': row['ethnicity.label']\n",
    "        })\n",
    "    else:\n",
    "        print(f\"skipping ethnicity for {idx} due to id = NaN\")\n",
    "\n",
    "    if not isinstance(row['geographicOrigin.id'], float):\n",
    "        individual.update(geographicOrigin={\n",
    "            'id': row['geographicOrigin.id'],\n",
    "            'label': row['geographicOrigin.label']\n",
    "        })\n",
    "    else:\n",
    "        print(f\"skipping geographicOrigin for {idx} due to id = NaN\")\n",
    "    \n",
    "    if not isinstance(row['diseases_diseaseCode.id'], float):\n",
    "        diseases_id_array = row['diseases_diseaseCode.id'].split(\",\")\n",
    "        diseases_label_array = row['diseases_diseaseCode.label'].split(\",\")\n",
    "        # assuming arrays have same length\n",
    "        assert len(diseases_id_array) == len(diseases_label_array), \"disease code and label don't match\"\n",
    "\n",
    "        diseases = []\n",
    "        for index, val in enumerate(diseases_id_array):\n",
    "            id_ = diseases_id_array[index]\n",
    "            label = diseases_label_array[index]\n",
    "            disease_code = {\n",
    "                \"id\": id_,\n",
    "                \"label\": label\n",
    "            }\n",
    "            disease = {}\n",
    "            disease.update(diseaseCode=disease_code)\n",
    "            diseases.append(disease)\n",
    "        if len(diseases) > 0:\n",
    "            individual.update(diseases=diseases)\n",
    "    else:\n",
    "        print(f\"skipping diseases for {idx} due to id = NaN\")\n",
    "        \n",
    "    if not isinstance(row['measures_assayCode.id'], float):\n",
    "        measures_assay_id_array = row['measures_assayCode.id'].split(\",\")\n",
    "        measures_assay_label_array = row['measures_assayCode.label'].split(\",\")\n",
    "        measures_date_array = row['measures_date'].split(\",\")\n",
    "        measures_measurementValue = ast.literal_eval(row['measures_measurementValue'])\n",
    "        # assuming arrays have same length\n",
    "        assert len(measures_date_array) == len(measures_measurementValue) == len(measures_assay_id_array) == len(measures_assay_label_array), \"measure assay code and label don't match\"\n",
    "\n",
    "        measures = []\n",
    "        for index, val in enumerate(measures_assay_id_array):\n",
    "            assay_code = {\n",
    "                'id': measures_assay_id_array[index],\n",
    "                'label': measures_assay_label_array[index]\n",
    "                }\n",
    "            measure = {}\n",
    "            measure.update(assayCode=assay_code)\n",
    "            measure.update(date=measures_date_array[index])\n",
    "            measure.update(measurementValue=measures_measurementValue[index])\n",
    "            measures.append(measure)\n",
    "        if len(measures) > 0:\n",
    "            individual.update(measures=measures)\n",
    "    else:\n",
    "        print(f\"skipping measures for {idx} due to id = NaN\")\n",
    "    \n",
    "    # add individual to dataset\n",
    "    submission_json[\"individuals\"].append(individual)\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {},
     "inputWidgets": {},
     "nuid": "dbdec490-4167-47df-bb79-0e340c945476",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/marionfs/jupyter-venv/lib/python3.8/site-packages/openpyxl/worksheet/_reader.py:312: UserWarning: Unknown extension is not supported and will be removed\n",
      "  warn(msg)\n"
     ]
    }
   ],
   "source": [
    "df_bs = pd.read_excel(f'{work_dir}/Beacon-v2-Models_CINECA_UK1MS.xlsx', sheet_name='biosamples')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {},
     "inputWidgets": {},
     "nuid": "ef085158-42c1-417e-8572-feb008655cca",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>biosampleStatus.id</th>\n",
       "      <th>biosampleStatus.label</th>\n",
       "      <th>collectionDate</th>\n",
       "      <th>collectionMoment</th>\n",
       "      <th>diagnosticMarkers_id</th>\n",
       "      <th>diagnosticMarkers_label</th>\n",
       "      <th>histologicalDiagnosis.id</th>\n",
       "      <th>histologicalDiagnosis.label</th>\n",
       "      <th>id</th>\n",
       "      <th>individualId</th>\n",
       "      <th>...</th>\n",
       "      <th>sampleOriginType.id</th>\n",
       "      <th>sampleOriginType.label</th>\n",
       "      <th>sampleProcessing.id</th>\n",
       "      <th>sampleProcessing.label</th>\n",
       "      <th>sampleStorage.id</th>\n",
       "      <th>sampleStorage.label</th>\n",
       "      <th>tumorGrade.id</th>\n",
       "      <th>tumorGrade.label</th>\n",
       "      <th>tumorProgression.id</th>\n",
       "      <th>tumorProgression.label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>EFO:0009654</td>\n",
       "      <td>reference sample</td>\n",
       "      <td>2020-09-11</td>\n",
       "      <td>P40Y1M1D</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HG00096</td>\n",
       "      <td>HG00096</td>\n",
       "      <td>...</td>\n",
       "      <td>UBERON:0000178</td>\n",
       "      <td>blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>EFO:0009654</td>\n",
       "      <td>reference sample</td>\n",
       "      <td>2020-09-11</td>\n",
       "      <td>P40Y1M1D</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HG00097</td>\n",
       "      <td>HG00097</td>\n",
       "      <td>...</td>\n",
       "      <td>UBERON:0000178</td>\n",
       "      <td>blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>EFO:0009654</td>\n",
       "      <td>reference sample</td>\n",
       "      <td>2020-09-11</td>\n",
       "      <td>P40Y1M1D</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HG00099</td>\n",
       "      <td>HG00099</td>\n",
       "      <td>...</td>\n",
       "      <td>UBERON:0000178</td>\n",
       "      <td>blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>EFO:0009654</td>\n",
       "      <td>reference sample</td>\n",
       "      <td>2020-09-11</td>\n",
       "      <td>P40Y1M1D</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HG00100</td>\n",
       "      <td>HG00100</td>\n",
       "      <td>...</td>\n",
       "      <td>UBERON:0000178</td>\n",
       "      <td>blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>EFO:0009654</td>\n",
       "      <td>reference sample</td>\n",
       "      <td>2020-09-11</td>\n",
       "      <td>P40Y1M1D</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HG00101</td>\n",
       "      <td>HG00101</td>\n",
       "      <td>...</td>\n",
       "      <td>UBERON:0000178</td>\n",
       "      <td>blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2499</th>\n",
       "      <td>EFO:0009654</td>\n",
       "      <td>reference sample</td>\n",
       "      <td>2020-09-11</td>\n",
       "      <td>P40Y1M1D</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NA21137</td>\n",
       "      <td>NA21137</td>\n",
       "      <td>...</td>\n",
       "      <td>UBERON:0000178</td>\n",
       "      <td>blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2500</th>\n",
       "      <td>EFO:0009654</td>\n",
       "      <td>reference sample</td>\n",
       "      <td>2020-09-11</td>\n",
       "      <td>P40Y1M1D</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NA21141</td>\n",
       "      <td>NA21141</td>\n",
       "      <td>...</td>\n",
       "      <td>UBERON:0000178</td>\n",
       "      <td>blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2501</th>\n",
       "      <td>EFO:0009654</td>\n",
       "      <td>reference sample</td>\n",
       "      <td>2020-09-11</td>\n",
       "      <td>P40Y1M1D</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NA21142</td>\n",
       "      <td>NA21142</td>\n",
       "      <td>...</td>\n",
       "      <td>UBERON:0000178</td>\n",
       "      <td>blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2502</th>\n",
       "      <td>EFO:0009654</td>\n",
       "      <td>reference sample</td>\n",
       "      <td>2020-09-11</td>\n",
       "      <td>P40Y1M1D</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NA21143</td>\n",
       "      <td>NA21143</td>\n",
       "      <td>...</td>\n",
       "      <td>UBERON:0000178</td>\n",
       "      <td>blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2503</th>\n",
       "      <td>EFO:0009654</td>\n",
       "      <td>reference sample</td>\n",
       "      <td>2020-09-11</td>\n",
       "      <td>P40Y1M1D</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NA21144</td>\n",
       "      <td>NA21144</td>\n",
       "      <td>...</td>\n",
       "      <td>UBERON:0000178</td>\n",
       "      <td>blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2504 rows × 65 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     biosampleStatus.id biosampleStatus.label collectionDate collectionMoment  \\\n",
       "0           EFO:0009654      reference sample     2020-09-11         P40Y1M1D   \n",
       "1           EFO:0009654      reference sample     2020-09-11         P40Y1M1D   \n",
       "2           EFO:0009654      reference sample     2020-09-11         P40Y1M1D   \n",
       "3           EFO:0009654      reference sample     2020-09-11         P40Y1M1D   \n",
       "4           EFO:0009654      reference sample     2020-09-11         P40Y1M1D   \n",
       "...                 ...                   ...            ...              ...   \n",
       "2499        EFO:0009654      reference sample     2020-09-11         P40Y1M1D   \n",
       "2500        EFO:0009654      reference sample     2020-09-11         P40Y1M1D   \n",
       "2501        EFO:0009654      reference sample     2020-09-11         P40Y1M1D   \n",
       "2502        EFO:0009654      reference sample     2020-09-11         P40Y1M1D   \n",
       "2503        EFO:0009654      reference sample     2020-09-11         P40Y1M1D   \n",
       "\n",
       "      diagnosticMarkers_id  diagnosticMarkers_label  histologicalDiagnosis.id  \\\n",
       "0                      NaN                      NaN                       NaN   \n",
       "1                      NaN                      NaN                       NaN   \n",
       "2                      NaN                      NaN                       NaN   \n",
       "3                      NaN                      NaN                       NaN   \n",
       "4                      NaN                      NaN                       NaN   \n",
       "...                    ...                      ...                       ...   \n",
       "2499                   NaN                      NaN                       NaN   \n",
       "2500                   NaN                      NaN                       NaN   \n",
       "2501                   NaN                      NaN                       NaN   \n",
       "2502                   NaN                      NaN                       NaN   \n",
       "2503                   NaN                      NaN                       NaN   \n",
       "\n",
       "      histologicalDiagnosis.label       id individualId  ...  \\\n",
       "0                             NaN  HG00096      HG00096  ...   \n",
       "1                             NaN  HG00097      HG00097  ...   \n",
       "2                             NaN  HG00099      HG00099  ...   \n",
       "3                             NaN  HG00100      HG00100  ...   \n",
       "4                             NaN  HG00101      HG00101  ...   \n",
       "...                           ...      ...          ...  ...   \n",
       "2499                          NaN  NA21137      NA21137  ...   \n",
       "2500                          NaN  NA21141      NA21141  ...   \n",
       "2501                          NaN  NA21142      NA21142  ...   \n",
       "2502                          NaN  NA21143      NA21143  ...   \n",
       "2503                          NaN  NA21144      NA21144  ...   \n",
       "\n",
       "     sampleOriginType.id sampleOriginType.label sampleProcessing.id  \\\n",
       "0         UBERON:0000178                  blood                 NaN   \n",
       "1         UBERON:0000178                  blood                 NaN   \n",
       "2         UBERON:0000178                  blood                 NaN   \n",
       "3         UBERON:0000178                  blood                 NaN   \n",
       "4         UBERON:0000178                  blood                 NaN   \n",
       "...                  ...                    ...                 ...   \n",
       "2499      UBERON:0000178                  blood                 NaN   \n",
       "2500      UBERON:0000178                  blood                 NaN   \n",
       "2501      UBERON:0000178                  blood                 NaN   \n",
       "2502      UBERON:0000178                  blood                 NaN   \n",
       "2503      UBERON:0000178                  blood                 NaN   \n",
       "\n",
       "     sampleProcessing.label sampleStorage.id  sampleStorage.label  \\\n",
       "0                       NaN              NaN                  NaN   \n",
       "1                       NaN              NaN                  NaN   \n",
       "2                       NaN              NaN                  NaN   \n",
       "3                       NaN              NaN                  NaN   \n",
       "4                       NaN              NaN                  NaN   \n",
       "...                     ...              ...                  ...   \n",
       "2499                    NaN              NaN                  NaN   \n",
       "2500                    NaN              NaN                  NaN   \n",
       "2501                    NaN              NaN                  NaN   \n",
       "2502                    NaN              NaN                  NaN   \n",
       "2503                    NaN              NaN                  NaN   \n",
       "\n",
       "      tumorGrade.id  tumorGrade.label  tumorProgression.id  \\\n",
       "0               NaN               NaN                  NaN   \n",
       "1               NaN               NaN                  NaN   \n",
       "2               NaN               NaN                  NaN   \n",
       "3               NaN               NaN                  NaN   \n",
       "4               NaN               NaN                  NaN   \n",
       "...             ...               ...                  ...   \n",
       "2499            NaN               NaN                  NaN   \n",
       "2500            NaN               NaN                  NaN   \n",
       "2501            NaN               NaN                  NaN   \n",
       "2502            NaN               NaN                  NaN   \n",
       "2503            NaN               NaN                  NaN   \n",
       "\n",
       "      tumorProgression.label  \n",
       "0                        NaN  \n",
       "1                        NaN  \n",
       "2                        NaN  \n",
       "3                        NaN  \n",
       "4                        NaN  \n",
       "...                      ...  \n",
       "2499                     NaN  \n",
       "2500                     NaN  \n",
       "2501                     NaN  \n",
       "2502                     NaN  \n",
       "2503                     NaN  \n",
       "\n",
       "[2504 rows x 65 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_bs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {},
     "inputWidgets": {},
     "nuid": "bb5f5149-d203-427f-b95c-a0648a6231d7",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['biosampleStatus.id', 'biosampleStatus.label', 'collectionDate',\n",
       "       'collectionMoment', 'diagnosticMarkers_id', 'diagnosticMarkers_label',\n",
       "       'histologicalDiagnosis.id', 'histologicalDiagnosis.label', 'id',\n",
       "       'individualId', 'info.BioSamples.accession',\n",
       "       'info.BioSamples.externalUrl', 'info.characteristics',\n",
       "       'info.EGAsampleId', 'info.sampleName', 'info.taxId',\n",
       "       'measurements_assayCode.id', 'measurements_assayCode.label',\n",
       "       'measurements_date', 'measurements_measurementValue',\n",
       "       'measurements_notes', 'measurements_observationMoment',\n",
       "       'measurements_procedure.ageAtProcedure',\n",
       "       'measurements_procedure.bodySite.id',\n",
       "       'measurements_procedure.bodySite.label',\n",
       "       'measurements_procedure.dateOfProcedure',\n",
       "       'measurements_procedure.procedureCode.id',\n",
       "       'measurements_procedure.procedureCode.label', 'notes',\n",
       "       'obtentionProcedure.ageAtProcedure', 'obtentionProcedure.bodySite.id',\n",
       "       'obtentionProcedure.bodySite.label',\n",
       "       'obtentionProcedure.dateOfProcedure',\n",
       "       'obtentionProcedure.procedureCode.id',\n",
       "       'obtentionProcedure.procedureCode.label', 'pathologicalStage.id',\n",
       "       'pathologicalStage.label', 'pathologicalTnmFinding_id',\n",
       "       'pathologicalTnmFinding_label',\n",
       "       'phenotypicFeatures_evidence.evidenceCode.id',\n",
       "       'phenotypicFeatures_evidence.evidenceCode.label',\n",
       "       'phenotypicFeatures_evidence.reference.id',\n",
       "       'phenotypicFeatures_evidence.reference.notes',\n",
       "       'phenotypicFeatures_evidence.reference.reference',\n",
       "       'phenotypicFeatures_excluded', 'phenotypicFeatures_featureType.id',\n",
       "       'phenotypicFeatures_featureType.label', 'phenotypicFeatures_modifiers',\n",
       "       'phenotypicFeatures_notes', 'phenotypicFeatures_onset',\n",
       "       'phenotypicFeatures_resolution', 'phenotypicFeatures_severity.id',\n",
       "       'phenotypicFeatures_severity.label', 'sampleOriginDetail.id',\n",
       "       'sampleOriginDetail.label', 'sampleOriginType.id',\n",
       "       'sampleOriginType.label', 'sampleProcessing.id',\n",
       "       'sampleProcessing.label', 'sampleStorage.id', 'sampleStorage.label',\n",
       "       'tumorGrade.id', 'tumorGrade.label', 'tumorProgression.id',\n",
       "       'tumorProgression.label'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_bs.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {},
     "inputWidgets": {},
     "nuid": "5badea52-c4f4-4d47-949f-e6d8a09246db",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [],
   "source": [
    "# add biosamples\n",
    "# randomly generate ages for some variety, based on BioBank min age 37 max age 73\n",
    "# adds blood draw and frozen specimen curations\n",
    "for idx, row in df_bs.iterrows():\n",
    "#for idx, row in df.head(1).iterrows():\n",
    "    # print(row)\n",
    "    \n",
    "    biosample = {}\n",
    "    biosample.update(id=row['id'])\n",
    "    biosample.update(individualId=row['id'])\n",
    "    # biosample.update(sampleName=row['info.sampleName'])\n",
    "    biosample.update(collectionDate=row['collectionDate'])\n",
    "    biosample.update(collectionMoment=f'P{random.randint(37,73)}Y')\n",
    "    biosample.update(sampleStorage={\n",
    "        'id': 'OBI:0000922',\n",
    "        'label': 'frozen specimen'})\n",
    "    biosample.update(obtentionProcedure={\n",
    "        'procedureCode': {'label': 'blood draw',\n",
    "                          'id': 'EFO:0009121'}})\n",
    "    \n",
    "    if not isinstance(row['biosampleStatus.id'], float):\n",
    "        biosample.update(biosampleStatus={\n",
    "            'id': row['biosampleStatus.id'],\n",
    "            'label': row['biosampleStatus.label']\n",
    "        })\n",
    "    else:\n",
    "        print(f\"ERROR: biosampleStatus for {idx} id = NaN\")\n",
    "        \n",
    "    if not isinstance(row['sampleOriginType.id'], float):\n",
    "        biosample.update(sampleOriginType={\n",
    "            'id': row['sampleOriginType.id'],\n",
    "            'label': row['sampleOriginType.label']\n",
    "        })\n",
    "    else:\n",
    "        print(f\"ERROR: sampleOriginType for {idx} id = NaN\")\n",
    "\n",
    "    submission_json[\"biosamples\"].append(biosample)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {},
     "inputWidgets": {},
     "nuid": "72df9b26-5b5c-4096-9832-09c5d7852451",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/marionfs/jupyter-venv/lib/python3.8/site-packages/openpyxl/worksheet/_reader.py:312: UserWarning: Unknown extension is not supported and will be removed\n",
      "  warn(msg)\n"
     ]
    }
   ],
   "source": [
    "df_an = pd.read_excel(f'{work_dir}/Beacon-v2-Models_CINECA_UK1MS.xlsx', sheet_name='analyses')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {},
     "inputWidgets": {},
     "nuid": "f2649ee6-ce47-4af3-a2f9-832fece8c02d",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>aligner</th>\n",
       "      <th>analysisDate</th>\n",
       "      <th>biosampleId</th>\n",
       "      <th>id</th>\n",
       "      <th>individualId</th>\n",
       "      <th>pipelineName</th>\n",
       "      <th>pipelineRef</th>\n",
       "      <th>runId</th>\n",
       "      <th>variantCaller</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>bwa-0.5.9</td>\n",
       "      <td>2015-09-30</td>\n",
       "      <td>HG00096</td>\n",
       "      <td>UK1_analysisId_1</td>\n",
       "      <td>HG00096</td>\n",
       "      <td>1000G-low-coverage-WGS</td>\n",
       "      <td>https://www.nature.com/articles/nature15393</td>\n",
       "      <td>SRR00000001</td>\n",
       "      <td>GATK4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>bwa-0.5.9</td>\n",
       "      <td>2015-09-30</td>\n",
       "      <td>HG00097</td>\n",
       "      <td>UK1_analysisId_2</td>\n",
       "      <td>HG00097</td>\n",
       "      <td>1000G-low-coverage-WGS</td>\n",
       "      <td>https://www.nature.com/articles/nature15393</td>\n",
       "      <td>SRR00000002</td>\n",
       "      <td>GATK4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>bwa-0.5.9</td>\n",
       "      <td>2015-09-30</td>\n",
       "      <td>HG00099</td>\n",
       "      <td>UK1_analysisId_3</td>\n",
       "      <td>HG00099</td>\n",
       "      <td>1000G-low-coverage-WGS</td>\n",
       "      <td>https://www.nature.com/articles/nature15393</td>\n",
       "      <td>SRR00000003</td>\n",
       "      <td>GATK4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>bwa-0.5.9</td>\n",
       "      <td>2015-09-30</td>\n",
       "      <td>HG00100</td>\n",
       "      <td>UK1_analysisId_4</td>\n",
       "      <td>HG00100</td>\n",
       "      <td>1000G-low-coverage-WGS</td>\n",
       "      <td>https://www.nature.com/articles/nature15393</td>\n",
       "      <td>SRR00000004</td>\n",
       "      <td>GATK4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>bwa-0.5.9</td>\n",
       "      <td>2015-09-30</td>\n",
       "      <td>HG00101</td>\n",
       "      <td>UK1_analysisId_5</td>\n",
       "      <td>HG00101</td>\n",
       "      <td>1000G-low-coverage-WGS</td>\n",
       "      <td>https://www.nature.com/articles/nature15393</td>\n",
       "      <td>SRR00000005</td>\n",
       "      <td>GATK4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2499</th>\n",
       "      <td>bwa-0.5.9</td>\n",
       "      <td>2015-09-30</td>\n",
       "      <td>NA21137</td>\n",
       "      <td>UK1_analysisId_2500</td>\n",
       "      <td>NA21137</td>\n",
       "      <td>1000G-low-coverage-WGS</td>\n",
       "      <td>https://www.nature.com/articles/nature15393</td>\n",
       "      <td>SRR00002500</td>\n",
       "      <td>GATK4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2500</th>\n",
       "      <td>bwa-0.5.9</td>\n",
       "      <td>2015-09-30</td>\n",
       "      <td>NA21141</td>\n",
       "      <td>UK1_analysisId_2501</td>\n",
       "      <td>NA21141</td>\n",
       "      <td>1000G-low-coverage-WGS</td>\n",
       "      <td>https://www.nature.com/articles/nature15393</td>\n",
       "      <td>SRR00002501</td>\n",
       "      <td>GATK4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2501</th>\n",
       "      <td>bwa-0.5.9</td>\n",
       "      <td>2015-09-30</td>\n",
       "      <td>NA21142</td>\n",
       "      <td>UK1_analysisId_2502</td>\n",
       "      <td>NA21142</td>\n",
       "      <td>1000G-low-coverage-WGS</td>\n",
       "      <td>https://www.nature.com/articles/nature15393</td>\n",
       "      <td>SRR00002502</td>\n",
       "      <td>GATK4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2502</th>\n",
       "      <td>bwa-0.5.9</td>\n",
       "      <td>2015-09-30</td>\n",
       "      <td>NA21143</td>\n",
       "      <td>UK1_analysisId_2503</td>\n",
       "      <td>NA21143</td>\n",
       "      <td>1000G-low-coverage-WGS</td>\n",
       "      <td>https://www.nature.com/articles/nature15393</td>\n",
       "      <td>SRR00002503</td>\n",
       "      <td>GATK4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2503</th>\n",
       "      <td>bwa-0.5.9</td>\n",
       "      <td>2015-09-30</td>\n",
       "      <td>NA21144</td>\n",
       "      <td>UK1_analysisId_2504</td>\n",
       "      <td>NA21144</td>\n",
       "      <td>1000G-low-coverage-WGS</td>\n",
       "      <td>https://www.nature.com/articles/nature15393</td>\n",
       "      <td>SRR00002504</td>\n",
       "      <td>GATK4.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2504 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        aligner analysisDate biosampleId                   id individualId  \\\n",
       "0     bwa-0.5.9   2015-09-30     HG00096     UK1_analysisId_1      HG00096   \n",
       "1     bwa-0.5.9   2015-09-30     HG00097     UK1_analysisId_2      HG00097   \n",
       "2     bwa-0.5.9   2015-09-30     HG00099     UK1_analysisId_3      HG00099   \n",
       "3     bwa-0.5.9   2015-09-30     HG00100     UK1_analysisId_4      HG00100   \n",
       "4     bwa-0.5.9   2015-09-30     HG00101     UK1_analysisId_5      HG00101   \n",
       "...         ...          ...         ...                  ...          ...   \n",
       "2499  bwa-0.5.9   2015-09-30     NA21137  UK1_analysisId_2500      NA21137   \n",
       "2500  bwa-0.5.9   2015-09-30     NA21141  UK1_analysisId_2501      NA21141   \n",
       "2501  bwa-0.5.9   2015-09-30     NA21142  UK1_analysisId_2502      NA21142   \n",
       "2502  bwa-0.5.9   2015-09-30     NA21143  UK1_analysisId_2503      NA21143   \n",
       "2503  bwa-0.5.9   2015-09-30     NA21144  UK1_analysisId_2504      NA21144   \n",
       "\n",
       "                pipelineName                                  pipelineRef  \\\n",
       "0     1000G-low-coverage-WGS  https://www.nature.com/articles/nature15393   \n",
       "1     1000G-low-coverage-WGS  https://www.nature.com/articles/nature15393   \n",
       "2     1000G-low-coverage-WGS  https://www.nature.com/articles/nature15393   \n",
       "3     1000G-low-coverage-WGS  https://www.nature.com/articles/nature15393   \n",
       "4     1000G-low-coverage-WGS  https://www.nature.com/articles/nature15393   \n",
       "...                      ...                                          ...   \n",
       "2499  1000G-low-coverage-WGS  https://www.nature.com/articles/nature15393   \n",
       "2500  1000G-low-coverage-WGS  https://www.nature.com/articles/nature15393   \n",
       "2501  1000G-low-coverage-WGS  https://www.nature.com/articles/nature15393   \n",
       "2502  1000G-low-coverage-WGS  https://www.nature.com/articles/nature15393   \n",
       "2503  1000G-low-coverage-WGS  https://www.nature.com/articles/nature15393   \n",
       "\n",
       "            runId variantCaller  \n",
       "0     SRR00000001       GATK4.0  \n",
       "1     SRR00000002       GATK4.0  \n",
       "2     SRR00000003       GATK4.0  \n",
       "3     SRR00000004       GATK4.0  \n",
       "4     SRR00000005       GATK4.0  \n",
       "...           ...           ...  \n",
       "2499  SRR00002500       GATK4.0  \n",
       "2500  SRR00002501       GATK4.0  \n",
       "2501  SRR00002502       GATK4.0  \n",
       "2502  SRR00002503       GATK4.0  \n",
       "2503  SRR00002504       GATK4.0  \n",
       "\n",
       "[2504 rows x 9 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_an"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {},
     "inputWidgets": {},
     "nuid": "0da542fe-b02c-41a1-8f40-6f35e7ceb8ec",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['aligner', 'analysisDate', 'biosampleId', 'id', 'individualId',\n",
       "       'pipelineName', 'pipelineRef', 'runId', 'variantCaller'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_an.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {},
     "inputWidgets": {},
     "nuid": "1a0521f7-ba24-41ef-990e-7b9fc82251f1",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [],
   "source": [
    "# add analysis\n",
    "for idx, row in df_an.iterrows():    \n",
    "    analysis = {}\n",
    "    analysis.update(id=row['id'])\n",
    "    analysis.update(runId=row['runId'])\n",
    "    analysis.update(individualId=row['individualId'])\n",
    "    analysis.update(biosampleId=row['biosampleId'])\n",
    "    analysis.update(analysisDate=row['analysisDate'])\n",
    "    analysis.update(pipelineName=row['pipelineName'])\n",
    "    analysis.update(vcfSampleId=row['individualId'])\n",
    "    \n",
    "    submission_json[\"analyses\"].append(analysis)\n",
    "    \n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {},
     "inputWidgets": {},
     "nuid": "260e0f89-8d0b-4946-84c6-04ecbbb14616",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/marionfs/jupyter-venv/lib/python3.8/site-packages/openpyxl/worksheet/_reader.py:312: UserWarning: Unknown extension is not supported and will be removed\n",
      "  warn(msg)\n"
     ]
    }
   ],
   "source": [
    "df_runs = pd.read_excel(f'{work_dir}/Beacon-v2-Models_CINECA_UK1MS.xlsx', sheet_name='runs')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {},
     "inputWidgets": {},
     "nuid": "83c2f21a-ca69-4493-b4c4-efe33c36936c",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>biosampleId</th>\n",
       "      <th>id</th>\n",
       "      <th>individualId</th>\n",
       "      <th>libraryLayout</th>\n",
       "      <th>librarySelection</th>\n",
       "      <th>librarySource.id</th>\n",
       "      <th>librarySource.label</th>\n",
       "      <th>libraryStrategy</th>\n",
       "      <th>platform</th>\n",
       "      <th>platformModel.id</th>\n",
       "      <th>platformModel.label</th>\n",
       "      <th>runDate</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HG00096</td>\n",
       "      <td>SRR00000001</td>\n",
       "      <td>HG00096</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>GENEPIO:0001966</td>\n",
       "      <td>genomic Source</td>\n",
       "      <td>Short read Illumina WGS sequencing</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>OBI:0002048</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>2015-09-30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HG00097</td>\n",
       "      <td>SRR00000002</td>\n",
       "      <td>HG00097</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>GENEPIO:0001966</td>\n",
       "      <td>genomic Source</td>\n",
       "      <td>Short read Illumina WGS sequencing</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>OBI:0002048</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>2015-09-30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HG00099</td>\n",
       "      <td>SRR00000003</td>\n",
       "      <td>HG00099</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>GENEPIO:0001966</td>\n",
       "      <td>genomic Source</td>\n",
       "      <td>Short read Illumina WGS sequencing</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>OBI:0002048</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>2015-09-30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HG00100</td>\n",
       "      <td>SRR00000004</td>\n",
       "      <td>HG00100</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>GENEPIO:0001966</td>\n",
       "      <td>genomic Source</td>\n",
       "      <td>Short read Illumina WGS sequencing</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>OBI:0002048</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>2015-09-30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HG00101</td>\n",
       "      <td>SRR00000005</td>\n",
       "      <td>HG00101</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>GENEPIO:0001966</td>\n",
       "      <td>genomic Source</td>\n",
       "      <td>Short read Illumina WGS sequencing</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>OBI:0002048</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>2015-09-30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2499</th>\n",
       "      <td>NA21137</td>\n",
       "      <td>SRR00002500</td>\n",
       "      <td>NA21137</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>GENEPIO:0001966</td>\n",
       "      <td>genomic Source</td>\n",
       "      <td>Short read Illumina WGS sequencing</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>OBI:0002048</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>2015-09-30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2500</th>\n",
       "      <td>NA21141</td>\n",
       "      <td>SRR00002501</td>\n",
       "      <td>NA21141</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>GENEPIO:0001966</td>\n",
       "      <td>genomic Source</td>\n",
       "      <td>Short read Illumina WGS sequencing</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>OBI:0002048</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>2015-09-30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2501</th>\n",
       "      <td>NA21142</td>\n",
       "      <td>SRR00002502</td>\n",
       "      <td>NA21142</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>GENEPIO:0001966</td>\n",
       "      <td>genomic Source</td>\n",
       "      <td>Short read Illumina WGS sequencing</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>OBI:0002048</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>2015-09-30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2502</th>\n",
       "      <td>NA21143</td>\n",
       "      <td>SRR00002503</td>\n",
       "      <td>NA21143</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>GENEPIO:0001966</td>\n",
       "      <td>genomic Source</td>\n",
       "      <td>Short read Illumina WGS sequencing</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>OBI:0002048</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>2015-09-30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2503</th>\n",
       "      <td>NA21144</td>\n",
       "      <td>SRR00002504</td>\n",
       "      <td>NA21144</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>GENEPIO:0001966</td>\n",
       "      <td>genomic Source</td>\n",
       "      <td>Short read Illumina WGS sequencing</td>\n",
       "      <td>Illumina</td>\n",
       "      <td>OBI:0002048</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>2015-09-30</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2504 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     biosampleId           id individualId libraryLayout librarySelection  \\\n",
       "0        HG00096  SRR00000001      HG00096        PAIRED           RANDOM   \n",
       "1        HG00097  SRR00000002      HG00097        PAIRED           RANDOM   \n",
       "2        HG00099  SRR00000003      HG00099        PAIRED           RANDOM   \n",
       "3        HG00100  SRR00000004      HG00100        PAIRED           RANDOM   \n",
       "4        HG00101  SRR00000005      HG00101        PAIRED           RANDOM   \n",
       "...          ...          ...          ...           ...              ...   \n",
       "2499     NA21137  SRR00002500      NA21137        PAIRED           RANDOM   \n",
       "2500     NA21141  SRR00002501      NA21141        PAIRED           RANDOM   \n",
       "2501     NA21142  SRR00002502      NA21142        PAIRED           RANDOM   \n",
       "2502     NA21143  SRR00002503      NA21143        PAIRED           RANDOM   \n",
       "2503     NA21144  SRR00002504      NA21144        PAIRED           RANDOM   \n",
       "\n",
       "     librarySource.id librarySource.label                     libraryStrategy  \\\n",
       "0     GENEPIO:0001966      genomic Source  Short read Illumina WGS sequencing   \n",
       "1     GENEPIO:0001966      genomic Source  Short read Illumina WGS sequencing   \n",
       "2     GENEPIO:0001966      genomic Source  Short read Illumina WGS sequencing   \n",
       "3     GENEPIO:0001966      genomic Source  Short read Illumina WGS sequencing   \n",
       "4     GENEPIO:0001966      genomic Source  Short read Illumina WGS sequencing   \n",
       "...               ...                 ...                                 ...   \n",
       "2499  GENEPIO:0001966      genomic Source  Short read Illumina WGS sequencing   \n",
       "2500  GENEPIO:0001966      genomic Source  Short read Illumina WGS sequencing   \n",
       "2501  GENEPIO:0001966      genomic Source  Short read Illumina WGS sequencing   \n",
       "2502  GENEPIO:0001966      genomic Source  Short read Illumina WGS sequencing   \n",
       "2503  GENEPIO:0001966      genomic Source  Short read Illumina WGS sequencing   \n",
       "\n",
       "      platform platformModel.id  platformModel.label     runDate  \n",
       "0     Illumina      OBI:0002048  Illumina HiSeq 2000  2015-09-30  \n",
       "1     Illumina      OBI:0002048  Illumina HiSeq 2000  2015-09-30  \n",
       "2     Illumina      OBI:0002048  Illumina HiSeq 2000  2015-09-30  \n",
       "3     Illumina      OBI:0002048  Illumina HiSeq 2000  2015-09-30  \n",
       "4     Illumina      OBI:0002048  Illumina HiSeq 2000  2015-09-30  \n",
       "...        ...              ...                  ...         ...  \n",
       "2499  Illumina      OBI:0002048  Illumina HiSeq 2000  2015-09-30  \n",
       "2500  Illumina      OBI:0002048  Illumina HiSeq 2000  2015-09-30  \n",
       "2501  Illumina      OBI:0002048  Illumina HiSeq 2000  2015-09-30  \n",
       "2502  Illumina      OBI:0002048  Illumina HiSeq 2000  2015-09-30  \n",
       "2503  Illumina      OBI:0002048  Illumina HiSeq 2000  2015-09-30  \n",
       "\n",
       "[2504 rows x 12 columns]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_runs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {},
     "inputWidgets": {},
     "nuid": "500fe579-202e-45cf-a92a-1ea1dcea2200",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [],
   "source": [
    "# add runs\n",
    "for idx, row in df_runs.iterrows():    \n",
    "    run = {}\n",
    "    run.update(id=row['id'])\n",
    "    run.update(individualId=row['individualId'])\n",
    "    run.update(biosampleId=row['biosampleId'])\n",
    "    run.update(runDate=row['runDate'])\n",
    "    run.update(libraryLayout=row['libraryLayout'])\n",
    "    run.update(librarySelection=row['librarySelection'])\n",
    "    run.update(libraryStrategy=row['libraryStrategy'])\n",
    "    run.update(platform=row['platform'])\n",
    "    \n",
    "    run.update(librarySource={\n",
    "            'id': row['librarySource.id'],\n",
    "            'label': row['librarySource.label']\n",
    "        })\n",
    "    run.update(platformModel={\n",
    "            'id': row['platformModel.id'],\n",
    "            'label': row['platformModel.label']\n",
    "        })\n",
    "\n",
    "    submission_json[\"runs\"].append(run)\n",
    "    \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Check linking between sheets\n",
    "\n",
    "All analysis.runId in runs.id\n",
    "All analysis.biosampleId in biosamples.id\n",
    "All anlaysis.individualId in individuals.id\n",
    "\n",
    "All biosample.individualId in individuals.id\n",
    "\n",
    "All runs.biosampleId in biosamples.id\n",
    "All runs.individualId in individuals.id\n",
    "\n",
    "How many runs per analysis\n",
    "How many biosamples per individual\n",
    "How many runs per biosample"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of runIds in analysis submission: 2504\n",
      "Number of unique runIds in analysis submission: 2504\n",
      "Number of ids in Runs: 2504\n",
      "Number of unique ids in Runs: 2504\n",
      "Number of intersecting ids analysis.runId - runs.id: 2504\n",
      "Number of ids not in both runs.id - analysis.runId: 0\n"
     ]
    }
   ],
   "source": [
    "analysis_run_ids = [x['runId'] for x in submission_json['analyses']]\n",
    "print(f'Number of runIds in analysis submission: {len(analysis_run_ids)}')\n",
    "analysis_run_ids_set = set(analysis_run_ids)\n",
    "print(f'Number of unique runIds in analysis submission: {len(analysis_run_ids_set)}')\n",
    "run_ids = [x['id'] for x in submission_json['runs']]\n",
    "run_ids_set = set(run_ids)\n",
    "print(f'Number of ids in Runs: {len(run_ids)}')\n",
    "print(f'Number of unique ids in Runs: {len(run_ids_set)}')\n",
    "\n",
    "print(f'Number of intersecting ids analysis.runId - runs.id: {len(analysis_run_ids_set.intersection(run_ids_set))}')\n",
    "print(f'Number of ids not in both runs.id - analysis.runId: {len(run_ids_set.symmetric_difference(analysis_run_ids_set))}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of biosampleIds in analysis submission: 2504\n",
      "Number of unique biosampleIds in analysis submission: 2504\n",
      "Number of ids in biosamples: 2504\n",
      "Number of unique ids in biosamples: 2504\n",
      "Number of intersecting ids analysis.biosampleId - biosamples.id: 2504\n",
      "Number of ids not in both biosample.id - analysis.biosampleId: 0\n"
     ]
    }
   ],
   "source": [
    "analysis_biosample_ids = [x['biosampleId'] for x in submission_json['analyses']]\n",
    "print(f'Number of biosampleIds in analysis submission: {len(analysis_biosample_ids)}')\n",
    "analysis_biosample_ids_set = set(analysis_biosample_ids)\n",
    "print(f'Number of unique biosampleIds in analysis submission: {len(analysis_biosample_ids_set)}')\n",
    "biosample_ids = [x['id'] for x in submission_json['biosamples']]\n",
    "biosample_ids_set = set(biosample_ids)\n",
    "print(f'Number of ids in biosamples: {len(biosample_ids)}')\n",
    "print(f'Number of unique ids in biosamples: {len(biosample_ids_set)}')\n",
    "\n",
    "print(f'Number of intersecting ids analysis.biosampleId - biosamples.id: {len(analysis_biosample_ids_set.intersection(biosample_ids_set))}')\n",
    "print(f'Number of ids not in both biosample.id - analysis.biosampleId: {len(biosample_ids_set.symmetric_difference(analysis_biosample_ids_set))}')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of individualIds in analysis submission: 2504\n",
      "Number of unique individualIds in analysis submission: 2504\n",
      "Number of ids in individuals: 2504\n",
      "Number of unique ids in individuals: 2504\n",
      "Number of intersecting ids analysis.individualId - individuals.id: 2504\n",
      "Number of ids not in both individual.id - analysis.individualId: 0\n"
     ]
    }
   ],
   "source": [
    "analysis_individual_ids = [x['individualId'] for x in submission_json['analyses']]\n",
    "print(f'Number of individualIds in analysis submission: {len(analysis_individual_ids)}')\n",
    "analysis_individual_ids_set = set(analysis_individual_ids)\n",
    "print(f'Number of unique individualIds in analysis submission: {len(analysis_individual_ids_set)}')\n",
    "individual_ids = [x['id'] for x in submission_json['individuals']]\n",
    "individual_ids_set = set(individual_ids)\n",
    "print(f'Number of ids in individuals: {len(individual_ids)}')\n",
    "print(f'Number of unique ids in individuals: {len(individual_ids_set)}')\n",
    "\n",
    "print(f'Number of intersecting ids analysis.individualId - individuals.id: {len(analysis_individual_ids_set.intersection(individual_ids_set))}')\n",
    "print(f'Number of ids not in both individual.id - analysis.individualId: {len(individual_ids_set.symmetric_difference(analysis_individual_ids_set))}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {},
     "inputWidgets": {},
     "nuid": "97159d1b-b838-4793-931d-951f9d3d18a9",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [],
   "source": [
    "file_dir = '/Users/marionfs/Documents/GitHub/sbeacon-exploration/scripts'\n",
    "with open(f'{file_dir}/submission_from_excel.json', 'w') as f:\n",
    "    f.write(json.dumps(submission_json, indent=4))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {},
     "inputWidgets": {},
     "nuid": "52a626f1-bb8f-47f9-82ea-d2ee201223b3",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'displayHTML' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Input \u001b[0;32mIn [33]\u001b[0m, in \u001b[0;36m<cell line: 2>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[38;5;66;03m#%python\u001b[39;00m\n\u001b[0;32m----> 2\u001b[0m \u001b[43mdisplayHTML\u001b[49m(\u001b[38;5;124m\"\"\"\u001b[39m\u001b[38;5;124m<a href=\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m/files/submission_from_excel.json\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m download>Download JOSN</a>\u001b[39m\u001b[38;5;124m\"\"\"\u001b[39m)\n",
      "\u001b[0;31mNameError\u001b[0m: name 'displayHTML' is not defined"
     ]
    }
   ],
   "source": [
    "#%python\n",
    "#displayHTML(\"\"\"<a href=\"/files/submission_from_excel.json\" download>Download JOSN</a>\"\"\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "application/vnd.databricks.v1+cell": {
     "cellMetadata": {},
     "inputWidgets": {},
     "nuid": "c1334518-19d5-4cd8-9de0-e7de2294d93b",
     "showTitle": false,
     "title": ""
    }
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "application/vnd.databricks.v1+notebook": {
   "dashboards": [],
   "language": "python",
   "notebookMetadata": {
    "mostRecentlyExecutedCommandWithImplicitDF": {
     "commandId": 1775781040040350,
     "dataframes": [
      "_sqldf"
     ]
    },
    "pythonIndentUnit": 4
   },
   "notebookName": "CINECA_UK1C",
   "notebookOrigID": 1775781040040349,
   "widgets": {}
  },
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.9"
  },
  "varInspector": {
   "cols": {
    "lenName": 16,
    "lenType": 16,
    "lenVar": 40
   },
   "kernels_config": {
    "python": {
     "delete_cmd_postfix": "",
     "delete_cmd_prefix": "del ",
     "library": "var_list.py",
     "varRefreshCmd": "print(var_dic_list())"
    },
    "r": {
     "delete_cmd_postfix": ") ",
     "delete_cmd_prefix": "rm(",
     "library": "var_list.r",
     "varRefreshCmd": "cat(var_dic_list()) "
    }
   },
   "position": {
    "height": "631.836px",
    "left": "1823.51px",
    "right": "20px",
    "top": "125px",
    "width": "349.473px"
   },
   "types_to_exclude": [
    "module",
    "function",
    "builtin_function_or_method",
    "instance",
    "_Feature"
   ],
   "window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
